<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 BL1 and BL2 | _main.knit</title>
  <meta name="description" content="<p>This is the code from the figures presented in the CopyKit manuscript -
Resolving clonal substructure from single cell genomic data using CopyKit.</p>" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="2 BL1 and BL2 | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is the code from the figures presented in the CopyKit manuscript -
Resolving clonal substructure from single cell genomic data using CopyKit.</p>" />
  <meta name="github-repo" content="navinlabcode/copykit" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 BL1 and BL2 | _main.knit" />
  
  <meta name="twitter:description" content="<p>This is the code from the figures presented in the CopyKit manuscript -
Resolving clonal substructure from single cell genomic data using CopyKit.</p>" />
  

<meta name="author" content="Darlan Conterno Minussi" />


<meta name="date" content="2022-02-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="bm1.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CopyKit Manuscript Code</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html"><i class="fa fa-check"></i><b>2</b> BL1 and BL2</a>
<ul>
<li class="chapter" data-level="2.1" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#bl1"><i class="fa fa-check"></i><b>2.1</b> BL1</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#filtering"><i class="fa fa-check"></i><b>2.1.1</b> Filtering</a></li>
<li class="chapter" data-level="2.1.2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#umap-and-clustering"><i class="fa fa-check"></i><b>2.1.2</b> UMAP and clustering</a></li>
<li class="chapter" data-level="2.1.3" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#subclones-heatmap"><i class="fa fa-check"></i><b>2.1.3</b> Subclones Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#bl2"><i class="fa fa-check"></i><b>2.2</b> BL2</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#filtering-1"><i class="fa fa-check"></i><b>2.2.1</b> Filtering</a></li>
<li class="chapter" data-level="2.2.2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#umap-and-clustering-1"><i class="fa fa-check"></i><b>2.2.2</b> UMAP and clustering</a></li>
<li class="chapter" data-level="2.2.3" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#subclones-heatmap-1"><i class="fa fa-check"></i><b>2.2.3</b> Subclones Heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="bm1.html"><a href="bm1.html"><i class="fa fa-check"></i><b>3</b> BM1</a>
<ul>
<li class="chapter" data-level="3.1" data-path="bm1.html"><a href="bm1.html#reading-datasets"><i class="fa fa-check"></i><b>3.1</b> Reading datasets</a></li>
<li class="chapter" data-level="3.2" data-path="bm1.html"><a href="bm1.html#running-umap-and-clustering"><i class="fa fa-check"></i><b>3.2</b> Running UMAP and Clustering</a></li>
<li class="chapter" data-level="3.3" data-path="bm1.html"><a href="bm1.html#consensus-tree"><i class="fa fa-check"></i><b>3.3</b> Consensus tree</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html"><i class="fa fa-check"></i><b>4</b> CM1 and CM2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1"><i class="fa fa-check"></i><b>4.1</b> CM1</a></li>
<li class="chapter" data-level="4.2" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1-colon"><i class="fa fa-check"></i><b>4.2</b> CM1 colon</a></li>
<li class="chapter" data-level="4.3" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1-liver"><i class="fa fa-check"></i><b>4.3</b> CM1 liver</a></li>
<li class="chapter" data-level="4.4" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1-merging"><i class="fa fa-check"></i><b>4.4</b> CM1 merging</a></li>
<li class="chapter" data-level="4.5" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2"><i class="fa fa-check"></i><b>4.5</b> CM2</a></li>
<li class="chapter" data-level="4.6" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2-colon"><i class="fa fa-check"></i><b>4.6</b> CM2 colon</a></li>
<li class="chapter" data-level="4.7" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2-liver"><i class="fa fa-check"></i><b>4.7</b> CM2 liver</a></li>
<li class="chapter" data-level="4.8" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2-merging"><i class="fa fa-check"></i><b>4.8</b> CM2 merging</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>5</b> Session Info</a></li>
<li class="chapter" data-level="6" data-path="copykit-paper.html"><a href="copykit-paper.html"><i class="fa fa-check"></i><b>6</b> CopyKit Paper</a>
<ul>
<li class="chapter" data-level="6.1" data-path="copykit-paper.html"><a href="copykit-paper.html#manuscript-code"><i class="fa fa-check"></i><b>6.1</b> Manuscript Code</a></li>
<li class="chapter" data-level="6.2" data-path="copykit-paper.html"><a href="copykit-paper.html#copykit-installation"><i class="fa fa-check"></i><b>6.2</b> CopyKit installation</a></li>
<li class="chapter" data-level="6.3" data-path="copykit-paper.html"><a href="copykit-paper.html#documentation"><i class="fa fa-check"></i><b>6.3</b> Documentation</a></li>
<li class="chapter" data-level="6.4" data-path="copykit-paper.html"><a href="copykit-paper.html#datasets"><i class="fa fa-check"></i><b>6.4</b> Datasets</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bl1-and-bl2" class="section level1" number="2">
<h1><span class="header-section-number">2</span> BL1 and BL2</h1>
<p>Setup</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb1-1"><a href="bl1-and-bl2.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(copykit))</span>
<span id="cb1-2"><a href="bl1-and-bl2.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-3"><a href="bl1-and-bl2.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstatix)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;rstatix&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     desc</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb5-1"><a href="bl1-and-bl2.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb5-2"><a href="bl1-and-bl2.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb5-3"><a href="bl1-and-bl2.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggnewscale)</span>
<span id="cb5-4"><a href="bl1-and-bl2.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;janitor&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:rstatix&#39;:
## 
##     make_clean_names</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     chisq.test, fisher.test</code></pre>
<p>Functions to Calculate Shannon Diversity Indexes</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb9-1"><a href="bl1-and-bl2.html#cb9-1" aria-hidden="true" tabindex="-1"></a>shan <span class="ot">&lt;-</span> <span class="cf">function</span>(data, indices) {</span>
<span id="cb9-2"><a href="bl1-and-bl2.html#cb9-2" aria-hidden="true" tabindex="-1"></a>  data_ind <span class="ot">&lt;-</span> data[indices]</span>
<span id="cb9-3"><a href="bl1-and-bl2.html#cb9-3" aria-hidden="true" tabindex="-1"></a>  prop <span class="ot">&lt;-</span> janitor<span class="sc">::</span><span class="fu">tabyl</span>(data_ind) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(percent)</span>
<span id="cb9-4"><a href="bl1-and-bl2.html#cb9-4" aria-hidden="true" tabindex="-1"></a>  div <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">sum</span>(prop<span class="sc">*</span><span class="fu">log</span>(prop))</span>
<span id="cb9-5"><a href="bl1-and-bl2.html#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(div)</span>
<span id="cb9-6"><a href="bl1-and-bl2.html#cb9-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-7"><a href="bl1-and-bl2.html#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="bl1-and-bl2.html#cb9-8" aria-hidden="true" tabindex="-1"></a>shan_sample <span class="ot">&lt;-</span> <span class="cf">function</span>(meta, group) {</span>
<span id="cb9-9"><a href="bl1-and-bl2.html#cb9-9" aria-hidden="true" tabindex="-1"></a>  meta_group <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span></span>
<span id="cb9-10"><a href="bl1-and-bl2.html#cb9-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(sectors <span class="sc">==</span> <span class="sc">!!</span>group)</span>
<span id="cb9-11"><a href="bl1-and-bl2.html#cb9-11" aria-hidden="true" tabindex="-1"></a>  group_perc <span class="ot">&lt;-</span> janitor<span class="sc">::</span><span class="fu">tabyl</span>(<span class="fu">as.character</span>(meta_group<span class="sc">$</span>subclones)) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(percent)</span>
<span id="cb9-12"><a href="bl1-and-bl2.html#cb9-12" aria-hidden="true" tabindex="-1"></a>  group_div <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">sum</span>(group_perc<span class="sc">*</span><span class="fu">log</span>(group_perc))</span>
<span id="cb9-13"><a href="bl1-and-bl2.html#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="bl1-and-bl2.html#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check for sectors composed of a unique subclone</span></span>
<span id="cb9-15"><a href="bl1-and-bl2.html#cb9-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (group_div <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb9-16"><a href="bl1-and-bl2.html#cb9-16" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">foci =</span> group,</span>
<span id="cb9-17"><a href="bl1-and-bl2.html#cb9-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">div =</span> <span class="dv">0</span>,</span>
<span id="cb9-18"><a href="bl1-and-bl2.html#cb9-18" aria-hidden="true" tabindex="-1"></a>                     <span class="at">lci =</span> <span class="dv">0</span>,</span>
<span id="cb9-19"><a href="bl1-and-bl2.html#cb9-19" aria-hidden="true" tabindex="-1"></a>                     <span class="at">uci =</span> <span class="dv">0</span>)</span>
<span id="cb9-20"><a href="bl1-and-bl2.html#cb9-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb9-21"><a href="bl1-and-bl2.html#cb9-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-22"><a href="bl1-and-bl2.html#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="bl1-and-bl2.html#cb9-23" aria-hidden="true" tabindex="-1"></a>  shan_group_boot <span class="ot">&lt;-</span></span>
<span id="cb9-24"><a href="bl1-and-bl2.html#cb9-24" aria-hidden="true" tabindex="-1"></a>    boot<span class="sc">::</span><span class="fu">boot</span>(</span>
<span id="cb9-25"><a href="bl1-and-bl2.html#cb9-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">as.character</span>(meta_group<span class="sc">$</span>subclones),</span>
<span id="cb9-26"><a href="bl1-and-bl2.html#cb9-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">statistic =</span> shan,</span>
<span id="cb9-27"><a href="bl1-and-bl2.html#cb9-27" aria-hidden="true" tabindex="-1"></a>      <span class="at">R =</span> <span class="dv">2000</span></span>
<span id="cb9-28"><a href="bl1-and-bl2.html#cb9-28" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb9-29"><a href="bl1-and-bl2.html#cb9-29" aria-hidden="true" tabindex="-1"></a>  shan_group_boot_ci <span class="ot">&lt;-</span> <span class="fu">suppressWarnings</span>(boot<span class="sc">::</span><span class="fu">boot.ci</span>(shan_group_boot))</span>
<span id="cb9-30"><a href="bl1-and-bl2.html#cb9-30" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">foci =</span> group,</span>
<span id="cb9-31"><a href="bl1-and-bl2.html#cb9-31" aria-hidden="true" tabindex="-1"></a>                   <span class="at">div =</span> group_div,</span>
<span id="cb9-32"><a href="bl1-and-bl2.html#cb9-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">lci =</span> shan_group_boot_ci<span class="sc">$</span>normal[<span class="dv">2</span>],</span>
<span id="cb9-33"><a href="bl1-and-bl2.html#cb9-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">uci =</span> shan_group_boot_ci<span class="sc">$</span>normal[<span class="dv">3</span>])</span>
<span id="cb9-34"><a href="bl1-and-bl2.html#cb9-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-35"><a href="bl1-and-bl2.html#cb9-35" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div id="bl1" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> BL1</h2>
<div id="filtering" class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Filtering</h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb10-1"><a href="bl1-and-bl2.html#cb10-1" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">runVarbin</span>(<span class="st">&quot;/mnt/lab/users/dminussi/projects/CopyKit_Manuscript_Code/datasets/BL1/marked_bams/&quot;</span>, <span class="at">remove_Y =</span> T)</span></code></pre></div>
<pre><code>## Counting reads for genome hg38 and resolution: 220kb</code></pre>
<pre><code>## 34 bam files had less than 10 mean bincounts and were removed.</code></pre>
<pre><code>## Performing GC correction.</code></pre>
<pre><code>## Smoothing bin counts.</code></pre>
<pre><code>## Running segmentation algorithm: CBS for genome hg38</code></pre>
<pre><code>## Merging levels.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb18-1"><a href="bl1-and-bl2.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding diploid cells and sub-setting the copykit object</span></span>
<span id="cb18-2"><a href="bl1-and-bl2.html#cb18-2" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">findAneuploidCells</span>(bl1)</span></code></pre></div>
<pre><code>## number of iterations= 28</code></pre>
<pre><code>## Copykit detected 607 that are possibly euploid cells using a resolution of: 0.069</code></pre>
<pre><code>## Added information to colData(CopyKit).</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb22-1"><a href="bl1-and-bl2.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl1, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;is_aneuploid&#39;</span>),</span>
<span id="cb22-2"><a href="bl1-and-bl2.html#cb22-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">row_split =</span> <span class="st">&#39;is_aneuploid&#39;</span>,</span>
<span id="cb22-3"><a href="bl1-and-bl2.html#cb22-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_threads =</span> <span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Ordering by consensus requires cluster information.</code></pre>
<pre><code>## Switching to hclust.</code></pre>
<pre><code>## No distance matrix detected in the scCNA object.</code></pre>
<pre><code>## Calculating distance matrix with metric: euclidean</code></pre>
<pre><code>## Using 50 cores.</code></pre>
<pre><code>## Access distance matrix with copykit::distMat()</code></pre>
<pre><code>## Done.</code></pre>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-2-1.png" width="768" /></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb31-1"><a href="bl1-and-bl2.html#cb31-1" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> bl1[,<span class="fu">colData</span>(bl1)<span class="sc">$</span>is_aneuploid <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb31-2"><a href="bl1-and-bl2.html#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="bl1-and-bl2.html#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding low-quality cells</span></span>
<span id="cb31-4"><a href="bl1-and-bl2.html#cb31-4" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">findOutliers</span>(bl1)</span></code></pre></div>
<pre><code>## Calculating correlation matrix.</code></pre>
<pre><code>## Marked 102 cells as outliers.</code></pre>
<pre><code>## Adding information to metadata. Access with colData(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb36-1"><a href="bl1-and-bl2.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl1, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;outlier&#39;</span>),</span>
<span id="cb36-2"><a href="bl1-and-bl2.html#cb36-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">row_split =</span> <span class="st">&#39;outlier&#39;</span>,</span>
<span id="cb36-3"><a href="bl1-and-bl2.html#cb36-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_threads =</span> <span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Ordering by consensus requires cluster information.</code></pre>
<pre><code>## Switching to hclust.</code></pre>
<pre><code>## No distance matrix detected in the scCNA object.</code></pre>
<pre><code>## Calculating distance matrix with metric: euclidean</code></pre>
<pre><code>## Using 50 cores.</code></pre>
<pre><code>## Access distance matrix with copykit::distMat()</code></pre>
<pre><code>## Done.</code></pre>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-2-2.png" width="768" /></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb45-1"><a href="bl1-and-bl2.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtaining metrics and plotting</span></span>
<span id="cb45-2"><a href="bl1-and-bl2.html#cb45-2" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">runMetrics</span>(bl1)</span></code></pre></div>
<pre><code>## Calculating overdispersion.</code></pre>
<pre><code>## Warning in selectChildren(jobs, timeout): error &#39;No child processes&#39; in select</code></pre>
<pre><code>## Counting breakpoints.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb50-1"><a href="bl1-and-bl2.html#cb50-1" aria-hidden="true" tabindex="-1"></a>bl1_p_metrics <span class="ot">&lt;-</span> <span class="fu">plotMetrics</span>(</span>
<span id="cb50-2"><a href="bl1-and-bl2.html#cb50-2" aria-hidden="true" tabindex="-1"></a>  bl1,</span>
<span id="cb50-3"><a href="bl1-and-bl2.html#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">metric =</span> <span class="fu">c</span>(</span>
<span id="cb50-4"><a href="bl1-and-bl2.html#cb50-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;reads_total&quot;</span>,</span>
<span id="cb50-5"><a href="bl1-and-bl2.html#cb50-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;percentage_duplicates&quot;</span>,</span>
<span id="cb50-6"><a href="bl1-and-bl2.html#cb50-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;overdispersion&quot;</span>,</span>
<span id="cb50-7"><a href="bl1-and-bl2.html#cb50-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;breakpoint_count&quot;</span></span>
<span id="cb50-8"><a href="bl1-and-bl2.html#cb50-8" aria-hidden="true" tabindex="-1"></a>  ), <span class="at">label =</span> <span class="st">&#39;outlier&#39;</span>,</span>
<span id="cb50-9"><a href="bl1-and-bl2.html#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">dodge.width =</span> <span class="fl">0.8</span>,</span>
<span id="cb50-10"><a href="bl1-and-bl2.html#cb50-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">2</span></span>
<span id="cb50-11"><a href="bl1-and-bl2.html#cb50-11" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;TRUE&quot;</span> <span class="ot">=</span> <span class="st">&quot;#DA614D&quot;</span>,</span>
<span id="cb50-12"><a href="bl1-and-bl2.html#cb50-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">&quot;FALSE&quot;</span> <span class="ot">=</span> <span class="st">&quot;#5F917A&quot;</span>))</span></code></pre></div>
<pre><code>## Coloring by: outlier</code></pre>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb53-1"><a href="bl1-and-bl2.html#cb53-1" aria-hidden="true" tabindex="-1"></a>bl1_p_metrics</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-2-3.png" width="768" /></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb54-1"><a href="bl1-and-bl2.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculating averages from the colData information and statistics</span></span>
<span id="cb54-2"><a href="bl1-and-bl2.html#cb54-2" aria-hidden="true" tabindex="-1"></a>bl1_metrics <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(bl1))</span>
<span id="cb54-3"><a href="bl1-and-bl2.html#cb54-3" aria-hidden="true" tabindex="-1"></a>bl1_metrics <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="bl1-and-bl2.html#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">reads_total_mean =</span> <span class="fu">mean</span>(reads_total),</span>
<span id="cb54-5"><a href="bl1-and-bl2.html#cb54-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">reads_total_sd =</span> <span class="fu">sd</span>(reads_total),</span>
<span id="cb54-6"><a href="bl1-and-bl2.html#cb54-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">percent_dup_mean =</span> <span class="fu">mean</span>(percentage_duplicates),</span>
<span id="cb54-7"><a href="bl1-and-bl2.html#cb54-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">percent_dup_sd =</span> <span class="fu">sd</span>(percentage_duplicates)</span>
<span id="cb54-8"><a href="bl1-and-bl2.html#cb54-8" aria-hidden="true" tabindex="-1"></a>            )</span></code></pre></div>
<pre><code>##   reads_total_mean reads_total_sd percent_dup_mean percent_dup_sd
## 1          1035783         246883       0.08147598    0.009734118</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb56-1"><a href="bl1-and-bl2.html#cb56-1" aria-hidden="true" tabindex="-1"></a>bl1_metrics <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-2"><a href="bl1-and-bl2.html#cb56-2" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">kruskal_test</span>(overdispersion <span class="sc">~</span> outlier)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 6
##   .y.                n statistic    df     p method        
## * &lt;chr&gt;          &lt;int&gt;     &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;         
## 1 overdispersion   895      3.95     1 0.047 Kruskal-Wallis</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb58-1"><a href="bl1-and-bl2.html#cb58-1" aria-hidden="true" tabindex="-1"></a>bl1_metrics <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb58-2"><a href="bl1-and-bl2.html#cb58-2" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">kruskal_test</span>(breakpoint_count <span class="sc">~</span> outlier)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 6
##   .y.                  n statistic    df        p method        
## * &lt;chr&gt;            &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;chr&gt;         
## 1 breakpoint_count   895      63.2     1 1.87e-15 Kruskal-Wallis</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb60-1"><a href="bl1-and-bl2.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># removing low-quality cells</span></span>
<span id="cb60-2"><a href="bl1-and-bl2.html#cb60-2" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> bl1[,<span class="fu">colData</span>(bl1)<span class="sc">$</span>outlier <span class="sc">==</span> <span class="cn">FALSE</span>]</span></code></pre></div>
</div>
<div id="umap-and-clustering" class="section level3" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> UMAP and clustering</h3>
<div class="sourceCode" id="cb61"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb61-1"><a href="bl1-and-bl2.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running UMAP and storing into the copykit object</span></span>
<span id="cb61-2"><a href="bl1-and-bl2.html#cb61-2" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(bl1)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb66-1"><a href="bl1-and-bl2.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Grid Search of Jaccard Similarity (cluster stability)</span></span>
<span id="cb66-2"><a href="bl1-and-bl2.html#cb66-2" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(bl1)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 10 with median jaccard similarity of: 1</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb70-1"><a href="bl1-and-bl2.html#cb70-1" aria-hidden="true" tabindex="-1"></a>bl1_primary_suggestedK_p <span class="ot">&lt;-</span> <span class="fu">plotSuggestedK</span>(bl1)</span>
<span id="cb70-2"><a href="bl1-and-bl2.html#cb70-2" aria-hidden="true" tabindex="-1"></a>bl1_primary_suggestedK_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl1_umap_cl-1.png" width="672" /></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb71-1"><a href="bl1-and-bl2.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding clusters using the suggested k value as per default settings, </span></span>
<span id="cb71-2"><a href="bl1-and-bl2.html#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="co"># can be changed with the argument k_subclones</span></span>
<span id="cb71-3"><a href="bl1-and-bl2.html#cb71-3" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(bl1)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 10</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 4 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb76-1"><a href="bl1-and-bl2.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting UMAP colored by subclones</span></span>
<span id="cb76-2"><a href="bl1-and-bl2.html#cb76-2" aria-hidden="true" tabindex="-1"></a>bl1_umap_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(bl1, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: subclones.</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb79-1"><a href="bl1-and-bl2.html#cb79-1" aria-hidden="true" tabindex="-1"></a>bl1_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl1_umap_cl-2.png" width="672" /></p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb80-1"><a href="bl1-and-bl2.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating consensus matrix and consensus phylo, </span></span>
<span id="cb80-2"><a href="bl1-and-bl2.html#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="co"># this will be used by copykit to order the subclones on the heatmap</span></span>
<span id="cb80-3"><a href="bl1-and-bl2.html#cb80-3" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(bl1)</span>
<span id="cb80-4"><a href="bl1-and-bl2.html#cb80-4" aria-hidden="true" tabindex="-1"></a>bl1 <span class="ot">&lt;-</span> <span class="fu">runConsensusPhylo</span>(bl1)</span></code></pre></div>
</div>
<div id="subclones-heatmap" class="section level3" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Subclones Heatmap</h3>
<div class="sourceCode" id="cb81"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb81-1"><a href="bl1-and-bl2.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting copy number heatmap and annotating with subclones</span></span>
<span id="cb81-2"><a href="bl1-and-bl2.html#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl1, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/bl1_subclones-1.png" width="768" /></p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb83-1"><a href="bl1-and-bl2.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding macro-spatial information from file names to colData</span></span>
<span id="cb83-2"><a href="bl1-and-bl2.html#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="co"># and renaming to (S)ectors</span></span>
<span id="cb83-3"><a href="bl1-and-bl2.html#cb83-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(bl1)<span class="sc">$</span>sectors <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(<span class="fu">colData</span>(bl1)<span class="sc">$</span>sample, <span class="st">&quot;(L[0-9]+L[0-9]+|L[0-9]+)&quot;</span>)</span>
<span id="cb83-4"><a href="bl1-and-bl2.html#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(bl1)<span class="sc">$</span>sectors <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(<span class="fu">colData</span>(bl1)<span class="sc">$</span>sectors,</span>
<span id="cb83-5"><a href="bl1-and-bl2.html#cb83-5" aria-hidden="true" tabindex="-1"></a>                                                    <span class="st">&quot;L&quot;</span>, <span class="st">&quot;S&quot;</span>)</span>
<span id="cb83-6"><a href="bl1-and-bl2.html#cb83-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-7"><a href="bl1-and-bl2.html#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the UMAP colored by sectors</span></span>
<span id="cb83-8"><a href="bl1-and-bl2.html#cb83-8" aria-hidden="true" tabindex="-1"></a>bl1_umap_sectors_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(bl1, <span class="at">label =</span> <span class="st">&#39;sectors&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: sectors.</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb86-1"><a href="bl1-and-bl2.html#cb86-1" aria-hidden="true" tabindex="-1"></a>bl1_umap_sectors_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl1_spatial_sections-1.png" width="672" /></p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb87-1"><a href="bl1-and-bl2.html#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating pie charts plots of the proportions of clones on each sector</span></span>
<span id="cb87-2"><a href="bl1-and-bl2.html#cb87-2" aria-hidden="true" tabindex="-1"></a>bl1_focis <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">colData</span>(bl1)<span class="sc">$</span>sectors)</span>
<span id="cb87-3"><a href="bl1-and-bl2.html#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="bl1-and-bl2.html#cb87-4" aria-hidden="true" tabindex="-1"></a>bl1_foci_pies <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb87-5"><a href="bl1-and-bl2.html#cb87-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-6"><a href="bl1-and-bl2.html#cb87-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(bl1_focis)) {</span>
<span id="cb87-7"><a href="bl1-and-bl2.html#cb87-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-8"><a href="bl1-and-bl2.html#cb87-8" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(bl1)) <span class="sc">%&gt;%</span></span>
<span id="cb87-9"><a href="bl1-and-bl2.html#cb87-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sectors <span class="sc">==</span> bl1_focis[i])</span>
<span id="cb87-10"><a href="bl1-and-bl2.html#cb87-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-11"><a href="bl1-and-bl2.html#cb87-11" aria-hidden="true" tabindex="-1"></a>  bl1_foci_pies[[i]] <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df) <span class="sc">+</span></span>
<span id="cb87-12"><a href="bl1-and-bl2.html#cb87-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> sectors, <span class="at">fill =</span> subclones),</span>
<span id="cb87-13"><a href="bl1-and-bl2.html#cb87-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">stat =</span> <span class="st">&#39;identity&#39;</span>) <span class="sc">+</span></span>
<span id="cb87-14"><a href="bl1-and-bl2.html#cb87-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb87-15"><a href="bl1-and-bl2.html#cb87-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">subclones_pal</span>(),</span>
<span id="cb87-16"><a href="bl1-and-bl2.html#cb87-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">limits =</span> force) <span class="sc">+</span></span>
<span id="cb87-17"><a href="bl1-and-bl2.html#cb87-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_polar</span>(<span class="at">theta =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb87-18"><a href="bl1-and-bl2.html#cb87-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(bl1_focis[i])</span>
<span id="cb87-19"><a href="bl1-and-bl2.html#cb87-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-20"><a href="bl1-and-bl2.html#cb87-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb87-21"><a href="bl1-and-bl2.html#cb87-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-22"><a href="bl1-and-bl2.html#cb87-22" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(bl1_foci_pies)</span></code></pre></div>
<p><img src="_main_files/figure-html/bl1_spatial_sections-2.png" width="672" /></p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb88-1"><a href="bl1-and-bl2.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting up a theme for the next plot</span></span>
<span id="cb88-2"><a href="bl1-and-bl2.html#cb88-2" aria-hidden="true" tabindex="-1"></a>my_theme <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb88-3"><a href="bl1-and-bl2.html#cb88-3" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb88-4"><a href="bl1-and-bl2.html#cb88-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> <span class="st">&quot;gray28&quot;</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb88-5"><a href="bl1-and-bl2.html#cb88-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-6"><a href="bl1-and-bl2.html#cb88-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-7"><a href="bl1-and-bl2.html#cb88-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> <span class="st">&quot;gray28&quot;</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb88-8"><a href="bl1-and-bl2.html#cb88-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-9"><a href="bl1-and-bl2.html#cb88-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-10"><a href="bl1-and-bl2.html#cb88-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-11"><a href="bl1-and-bl2.html#cb88-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb88-12"><a href="bl1-and-bl2.html#cb88-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>)</span>
<span id="cb88-13"><a href="bl1-and-bl2.html#cb88-13" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb88-14"><a href="bl1-and-bl2.html#cb88-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;umap1&quot;</span>),</span>
<span id="cb88-15"><a href="bl1-and-bl2.html#cb88-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;umap2&quot;</span>)</span>
<span id="cb88-16"><a href="bl1-and-bl2.html#cb88-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb88-17"><a href="bl1-and-bl2.html#cb88-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-18"><a href="bl1-and-bl2.html#cb88-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the UMAP colored by sector and using ggrepel to label the subclones</span></span>
<span id="cb88-19"><a href="bl1-and-bl2.html#cb88-19" aria-hidden="true" tabindex="-1"></a>bl1_lfoci_umap <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">reducedDim</span>(bl1, <span class="st">&#39;umap&#39;</span>))</span>
<span id="cb88-20"><a href="bl1-and-bl2.html#cb88-20" aria-hidden="true" tabindex="-1"></a>bl1_lfoci_umap<span class="sc">$</span>sectors <span class="ot">&lt;-</span> <span class="fu">colData</span>(bl1)<span class="sc">$</span>sectors</span>
<span id="cb88-21"><a href="bl1-and-bl2.html#cb88-21" aria-hidden="true" tabindex="-1"></a>bl1_lfoci_umap<span class="sc">$</span>subclones <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">colData</span>(bl1)<span class="sc">$</span>subclones)</span>
<span id="cb88-22"><a href="bl1-and-bl2.html#cb88-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Avoiding overplotting of text</span></span>
<span id="cb88-23"><a href="bl1-and-bl2.html#cb88-23" aria-hidden="true" tabindex="-1"></a>bl1_lfoci_umap<span class="sc">$</span>subclones <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="sc">!</span><span class="fu">duplicated</span>(bl1_lfoci_umap<span class="sc">$</span>subclones),</span>
<span id="cb88-24"><a href="bl1-and-bl2.html#cb88-24" aria-hidden="true" tabindex="-1"></a>                                   bl1_lfoci_umap<span class="sc">$</span>subclones, <span class="st">&quot;&quot;</span>)</span>
<span id="cb88-25"><a href="bl1-and-bl2.html#cb88-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-26"><a href="bl1-and-bl2.html#cb88-26" aria-hidden="true" tabindex="-1"></a>bl1_lfoci_umap_p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(bl1_lfoci_umap, <span class="fu">aes</span>(V1, V2, <span class="at">label =</span> subclones)) <span class="sc">+</span></span>
<span id="cb88-27"><a href="bl1-and-bl2.html#cb88-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill =</span> sectors), <span class="at">shape =</span> <span class="dv">21</span>,</span>
<span id="cb88-28"><a href="bl1-and-bl2.html#cb88-28" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke =</span> <span class="fl">0.1</span>,</span>
<span id="cb88-29"><a href="bl1-and-bl2.html#cb88-29" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb88-30"><a href="bl1-and-bl2.html#cb88-30" aria-hidden="true" tabindex="-1"></a>  ggnewscale<span class="sc">::</span><span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb88-31"><a href="bl1-and-bl2.html#cb88-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">color =</span> subclones),</span>
<span id="cb88-32"><a href="bl1-and-bl2.html#cb88-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">min.segment.length =</span> <span class="dv">0</span>,</span>
<span id="cb88-33"><a href="bl1-and-bl2.html#cb88-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">box.padding =</span> <span class="fl">0.01</span>,</span>
<span id="cb88-34"><a href="bl1-and-bl2.html#cb88-34" aria-hidden="true" tabindex="-1"></a>                   <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb88-35"><a href="bl1-and-bl2.html#cb88-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb88-36"><a href="bl1-and-bl2.html#cb88-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">na.rm =</span> T) <span class="sc">+</span></span>
<span id="cb88-37"><a href="bl1-and-bl2.html#cb88-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">subclones_pal</span>(),</span>
<span id="cb88-38"><a href="bl1-and-bl2.html#cb88-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">limits =</span> force) <span class="sc">+</span></span>
<span id="cb88-39"><a href="bl1-and-bl2.html#cb88-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb88-40"><a href="bl1-and-bl2.html#cb88-40" aria-hidden="true" tabindex="-1"></a>  my_theme</span>
<span id="cb88-41"><a href="bl1-and-bl2.html#cb88-41" aria-hidden="true" tabindex="-1"></a>bl1_lfoci_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl1_spatial_sections-3.png" width="672" /></p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb89-1"><a href="bl1-and-bl2.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting consensus heatmap with a frequency annotation of sectors by </span></span>
<span id="cb89-2"><a href="bl1-and-bl2.html#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="co"># subclone and annotating genes of interest</span></span>
<span id="cb89-3"><a href="bl1-and-bl2.html#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl1, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">consensus =</span> T, <span class="at">group =</span> <span class="st">&#39;sectors&#39;</span>,</span>
<span id="cb89-4"><a href="bl1-and-bl2.html#cb89-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">genes =</span> <span class="fu">c</span>(</span>
<span id="cb89-5"><a href="bl1-and-bl2.html#cb89-5" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;FHIT&quot;</span>,</span>
<span id="cb89-6"><a href="bl1-and-bl2.html#cb89-6" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;CUX1&quot;</span>,</span>
<span id="cb89-7"><a href="bl1-and-bl2.html#cb89-7" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;WWOX&quot;</span>,</span>
<span id="cb89-8"><a href="bl1-and-bl2.html#cb89-8" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;FOXO1&quot;</span>,</span>
<span id="cb89-9"><a href="bl1-and-bl2.html#cb89-9" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;APC&quot;</span>,</span>
<span id="cb89-10"><a href="bl1-and-bl2.html#cb89-10" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;BCL2&quot;</span>,</span>
<span id="cb89-11"><a href="bl1-and-bl2.html#cb89-11" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;KRAS&quot;</span>,</span>
<span id="cb89-12"><a href="bl1-and-bl2.html#cb89-12" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;PGR&quot;</span>,</span>
<span id="cb89-13"><a href="bl1-and-bl2.html#cb89-13" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;PDGFRA&quot;</span>,</span>
<span id="cb89-14"><a href="bl1-and-bl2.html#cb89-14" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;ROS1&quot;</span>,</span>
<span id="cb89-15"><a href="bl1-and-bl2.html#cb89-15" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;PTPN11&quot;</span>,</span>
<span id="cb89-16"><a href="bl1-and-bl2.html#cb89-16" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb89-17"><a href="bl1-and-bl2.html#cb89-17" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;BTG1&quot;</span>,</span>
<span id="cb89-18"><a href="bl1-and-bl2.html#cb89-18" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;FGFR3&quot;</span>,</span>
<span id="cb89-19"><a href="bl1-and-bl2.html#cb89-19" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;PTEN&quot;</span>,</span>
<span id="cb89-20"><a href="bl1-and-bl2.html#cb89-20" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;FGFR2&quot;</span></span>
<span id="cb89-21"><a href="bl1-and-bl2.html#cb89-21" aria-hidden="true" tabindex="-1"></a>            ))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/bl1_spatial_sections-4.png" width="672" /></p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb91-1"><a href="bl1-and-bl2.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating Diversity Indexes by sector</span></span>
<span id="cb91-2"><a href="bl1-and-bl2.html#cb91-2" aria-hidden="true" tabindex="-1"></a>bl1_meta <span class="ot">&lt;-</span> <span class="fu">colData</span>(bl1)</span>
<span id="cb91-3"><a href="bl1-and-bl2.html#cb91-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-4"><a href="bl1-and-bl2.html#cb91-4" aria-hidden="true" tabindex="-1"></a>bl1_meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(bl1_meta)</span>
<span id="cb91-5"><a href="bl1-and-bl2.html#cb91-5" aria-hidden="true" tabindex="-1"></a>bl1_meta <span class="ot">&lt;-</span> bl1_meta[<span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>, <span class="st">&#39;sectors&#39;</span>)]</span>
<span id="cb91-6"><a href="bl1-and-bl2.html#cb91-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-7"><a href="bl1-and-bl2.html#cb91-7" aria-hidden="true" tabindex="-1"></a>bl1_L1_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl1_meta, <span class="st">&quot;S1&quot;</span>)</span>
<span id="cb91-8"><a href="bl1-and-bl2.html#cb91-8" aria-hidden="true" tabindex="-1"></a>bl1_L2_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl1_meta, <span class="st">&quot;S2&quot;</span>)</span>
<span id="cb91-9"><a href="bl1-and-bl2.html#cb91-9" aria-hidden="true" tabindex="-1"></a>bl1_L3_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl1_meta, <span class="st">&quot;S3&quot;</span>)</span>
<span id="cb91-10"><a href="bl1-and-bl2.html#cb91-10" aria-hidden="true" tabindex="-1"></a>bl1_L4L5_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl1_meta, <span class="st">&quot;S4S5&quot;</span>)</span>
<span id="cb91-11"><a href="bl1-and-bl2.html#cb91-11" aria-hidden="true" tabindex="-1"></a>bl1_L6L7_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl1_meta, <span class="st">&quot;S6S7&quot;</span>)</span>
<span id="cb91-12"><a href="bl1-and-bl2.html#cb91-12" aria-hidden="true" tabindex="-1"></a>bl1_L8L9_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl1_meta, <span class="st">&quot;S8S9&quot;</span>)</span>
<span id="cb91-13"><a href="bl1-and-bl2.html#cb91-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-14"><a href="bl1-and-bl2.html#cb91-14" aria-hidden="true" tabindex="-1"></a>bl1_div <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb91-15"><a href="bl1-and-bl2.html#cb91-15" aria-hidden="true" tabindex="-1"></a>  bl1_L1_div,</span>
<span id="cb91-16"><a href="bl1-and-bl2.html#cb91-16" aria-hidden="true" tabindex="-1"></a>  bl1_L2_div,</span>
<span id="cb91-17"><a href="bl1-and-bl2.html#cb91-17" aria-hidden="true" tabindex="-1"></a>  bl1_L3_div,</span>
<span id="cb91-18"><a href="bl1-and-bl2.html#cb91-18" aria-hidden="true" tabindex="-1"></a>  bl1_L4L5_div,</span>
<span id="cb91-19"><a href="bl1-and-bl2.html#cb91-19" aria-hidden="true" tabindex="-1"></a>  bl1_L6L7_div,</span>
<span id="cb91-20"><a href="bl1-and-bl2.html#cb91-20" aria-hidden="true" tabindex="-1"></a>  bl1_L8L9_div</span>
<span id="cb91-21"><a href="bl1-and-bl2.html#cb91-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb91-22"><a href="bl1-and-bl2.html#cb91-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-23"><a href="bl1-and-bl2.html#cb91-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting Shannon Diversity Index</span></span>
<span id="cb91-24"><a href="bl1-and-bl2.html#cb91-24" aria-hidden="true" tabindex="-1"></a>bl1_div_p <span class="ot">&lt;-</span> bl1_div <span class="sc">%&gt;%</span></span>
<span id="cb91-25"><a href="bl1-and-bl2.html#cb91-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb91-26"><a href="bl1-and-bl2.html#cb91-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(</span>
<span id="cb91-27"><a href="bl1-and-bl2.html#cb91-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">fct_reorder</span>(foci, div),</span>
<span id="cb91-28"><a href="bl1-and-bl2.html#cb91-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymin =</span> lci,</span>
<span id="cb91-29"><a href="bl1-and-bl2.html#cb91-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymax =</span> uci</span>
<span id="cb91-30"><a href="bl1-and-bl2.html#cb91-30" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb91-31"><a href="bl1-and-bl2.html#cb91-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> .<span class="dv">1</span>,</span>
<span id="cb91-32"><a href="bl1-and-bl2.html#cb91-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb91-33"><a href="bl1-and-bl2.html#cb91-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb91-34"><a href="bl1-and-bl2.html#cb91-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">fct_reorder</span>(foci, div),</span>
<span id="cb91-35"><a href="bl1-and-bl2.html#cb91-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> div,</span>
<span id="cb91-36"><a href="bl1-and-bl2.html#cb91-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> foci</span>
<span id="cb91-37"><a href="bl1-and-bl2.html#cb91-37" aria-hidden="true" tabindex="-1"></a>  ), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb91-38"><a href="bl1-and-bl2.html#cb91-38" aria-hidden="true" tabindex="-1"></a>  cowplot<span class="sc">::</span><span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb91-39"><a href="bl1-and-bl2.html#cb91-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb91-40"><a href="bl1-and-bl2.html#cb91-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(</span>
<span id="cb91-41"><a href="bl1-and-bl2.html#cb91-41" aria-hidden="true" tabindex="-1"></a>          <span class="at">angle =</span> <span class="dv">90</span>,</span>
<span id="cb91-42"><a href="bl1-and-bl2.html#cb91-42" aria-hidden="true" tabindex="-1"></a>          <span class="at">vjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb91-43"><a href="bl1-and-bl2.html#cb91-43" aria-hidden="true" tabindex="-1"></a>          <span class="at">hjust =</span> <span class="dv">1</span></span>
<span id="cb91-44"><a href="bl1-and-bl2.html#cb91-44" aria-hidden="true" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb91-45"><a href="bl1-and-bl2.html#cb91-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> scales<span class="sc">::</span><span class="fu">pretty_breaks</span>(<span class="at">n =</span> <span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb91-46"><a href="bl1-and-bl2.html#cb91-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;shannon diversity index&quot;</span>) <span class="sc">+</span></span>
<span id="cb91-47"><a href="bl1-and-bl2.html#cb91-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb91-48"><a href="bl1-and-bl2.html#cb91-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb91-49"><a href="bl1-and-bl2.html#cb91-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-50"><a href="bl1-and-bl2.html#cb91-50" aria-hidden="true" tabindex="-1"></a>bl1_div_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl1_spatial_sections-5.png" width="672" /></p>
</div>
</div>
<div id="bl2" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> BL2</h2>
<div id="filtering-1" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Filtering</h3>
<div class="sourceCode" id="cb92"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb92-1"><a href="bl1-and-bl2.html#cb92-1" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">runVarbin</span>(<span class="st">&quot;/mnt/lab/users/dminussi/projects/CopyKit_Manuscript_Code/datasets/BL2/marked_bams/&quot;</span>, <span class="at">remove_Y =</span> T)</span></code></pre></div>
<pre><code>## Counting reads for genome hg38 and resolution: 220kb</code></pre>
<pre><code>## 61 bam files had less than 10 mean bincounts and were removed.</code></pre>
<pre><code>## Performing GC correction.</code></pre>
<pre><code>## Smoothing bin counts.</code></pre>
<pre><code>## Running segmentation algorithm: CBS for genome hg38</code></pre>
<pre><code>## Merging levels.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb100"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb100-1"><a href="bl1-and-bl2.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding diploid cells and sub-setting the copykit object</span></span>
<span id="cb100-2"><a href="bl1-and-bl2.html#cb100-2" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">findAneuploidCells</span>(bl2)</span></code></pre></div>
<pre><code>## number of iterations= 24</code></pre>
<pre><code>## Copykit detected 100 that are possibly euploid cells using a resolution of: 0.068</code></pre>
<pre><code>## Added information to colData(CopyKit).</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb104-1"><a href="bl1-and-bl2.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl2, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;is_aneuploid&#39;</span>),</span>
<span id="cb104-2"><a href="bl1-and-bl2.html#cb104-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">row_split =</span> <span class="st">&#39;is_aneuploid&#39;</span>,</span>
<span id="cb104-3"><a href="bl1-and-bl2.html#cb104-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_threads =</span> <span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Ordering by consensus requires cluster information.</code></pre>
<pre><code>## Switching to hclust.</code></pre>
<pre><code>## No distance matrix detected in the scCNA object.</code></pre>
<pre><code>## Calculating distance matrix with metric: euclidean</code></pre>
<pre><code>## Using 50 cores.</code></pre>
<pre><code>## Access distance matrix with copykit::distMat()</code></pre>
<pre><code>## Done.</code></pre>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-3-1.png" width="768" /></p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb113-1"><a href="bl1-and-bl2.html#cb113-1" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> bl2[,<span class="fu">colData</span>(bl2)<span class="sc">$</span>is_aneuploid <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb113-2"><a href="bl1-and-bl2.html#cb113-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-3"><a href="bl1-and-bl2.html#cb113-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding low-quality cells</span></span>
<span id="cb113-4"><a href="bl1-and-bl2.html#cb113-4" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">findOutliers</span>(bl2)</span></code></pre></div>
<pre><code>## Calculating correlation matrix.</code></pre>
<pre><code>## Marked 108 cells as outliers.</code></pre>
<pre><code>## Adding information to metadata. Access with colData(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb118-1"><a href="bl1-and-bl2.html#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl2, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;outlier&#39;</span>),</span>
<span id="cb118-2"><a href="bl1-and-bl2.html#cb118-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">row_split =</span> <span class="st">&#39;outlier&#39;</span>,</span>
<span id="cb118-3"><a href="bl1-and-bl2.html#cb118-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_threads =</span> <span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Ordering by consensus requires cluster information.</code></pre>
<pre><code>## Switching to hclust.</code></pre>
<pre><code>## No distance matrix detected in the scCNA object.</code></pre>
<pre><code>## Calculating distance matrix with metric: euclidean</code></pre>
<pre><code>## Using 50 cores.</code></pre>
<pre><code>## Access distance matrix with copykit::distMat()</code></pre>
<pre><code>## Done.</code></pre>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-3-2.png" width="768" /></p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb127-1"><a href="bl1-and-bl2.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtaining metrics and plotting</span></span>
<span id="cb127-2"><a href="bl1-and-bl2.html#cb127-2" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">runMetrics</span>(bl2)</span></code></pre></div>
<pre><code>## Calculating overdispersion.</code></pre>
<pre><code>## Counting breakpoints.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb131"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb131-1"><a href="bl1-and-bl2.html#cb131-1" aria-hidden="true" tabindex="-1"></a>bl2_p_metrics <span class="ot">&lt;-</span> <span class="fu">plotMetrics</span>(</span>
<span id="cb131-2"><a href="bl1-and-bl2.html#cb131-2" aria-hidden="true" tabindex="-1"></a>  bl2,</span>
<span id="cb131-3"><a href="bl1-and-bl2.html#cb131-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">metric =</span> <span class="fu">c</span>(</span>
<span id="cb131-4"><a href="bl1-and-bl2.html#cb131-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;reads_total&quot;</span>,</span>
<span id="cb131-5"><a href="bl1-and-bl2.html#cb131-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;percentage_duplicates&quot;</span>,</span>
<span id="cb131-6"><a href="bl1-and-bl2.html#cb131-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;overdispersion&quot;</span>,</span>
<span id="cb131-7"><a href="bl1-and-bl2.html#cb131-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;breakpoint_count&quot;</span></span>
<span id="cb131-8"><a href="bl1-and-bl2.html#cb131-8" aria-hidden="true" tabindex="-1"></a>  ), <span class="at">label =</span> <span class="st">&#39;outlier&#39;</span>,</span>
<span id="cb131-9"><a href="bl1-and-bl2.html#cb131-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">dodge.width =</span> <span class="fl">0.8</span>,</span>
<span id="cb131-10"><a href="bl1-and-bl2.html#cb131-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">2</span></span>
<span id="cb131-11"><a href="bl1-and-bl2.html#cb131-11" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;TRUE&quot;</span> <span class="ot">=</span> <span class="st">&quot;#DA614D&quot;</span>,</span>
<span id="cb131-12"><a href="bl1-and-bl2.html#cb131-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">&quot;FALSE&quot;</span> <span class="ot">=</span> <span class="st">&quot;#5F917A&quot;</span>))</span></code></pre></div>
<pre><code>## Coloring by: outlier</code></pre>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb134"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb134-1"><a href="bl1-and-bl2.html#cb134-1" aria-hidden="true" tabindex="-1"></a>bl2_p_metrics</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-3-3.png" width="768" /></p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb135-1"><a href="bl1-and-bl2.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculating averages from the colData information and statistics</span></span>
<span id="cb135-2"><a href="bl1-and-bl2.html#cb135-2" aria-hidden="true" tabindex="-1"></a>bl2_metrics <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(bl2))</span>
<span id="cb135-3"><a href="bl1-and-bl2.html#cb135-3" aria-hidden="true" tabindex="-1"></a>bl2_metrics <span class="sc">%&gt;%</span></span>
<span id="cb135-4"><a href="bl1-and-bl2.html#cb135-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">reads_total_mean =</span> <span class="fu">mean</span>(reads_total),</span>
<span id="cb135-5"><a href="bl1-and-bl2.html#cb135-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">reads_total_sd =</span> <span class="fu">sd</span>(reads_total),</span>
<span id="cb135-6"><a href="bl1-and-bl2.html#cb135-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">percent_dup_mean =</span> <span class="fu">mean</span>(percentage_duplicates),</span>
<span id="cb135-7"><a href="bl1-and-bl2.html#cb135-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">percent_dup_sd =</span> <span class="fu">sd</span>(percentage_duplicates)</span>
<span id="cb135-8"><a href="bl1-and-bl2.html#cb135-8" aria-hidden="true" tabindex="-1"></a>            )</span></code></pre></div>
<pre><code>##   reads_total_mean reads_total_sd percent_dup_mean percent_dup_sd
## 1         975057.5       303911.9       0.08932727     0.01065848</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb137-1"><a href="bl1-and-bl2.html#cb137-1" aria-hidden="true" tabindex="-1"></a>bl2_metrics <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb137-2"><a href="bl1-and-bl2.html#cb137-2" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">kruskal_test</span>(overdispersion <span class="sc">~</span> outlier)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 6
##   .y.                n statistic    df             p method        
## * &lt;chr&gt;          &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt;         
## 1 overdispersion  1375      32.9     1 0.00000000978 Kruskal-Wallis</code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb139-1"><a href="bl1-and-bl2.html#cb139-1" aria-hidden="true" tabindex="-1"></a>bl2_metrics <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb139-2"><a href="bl1-and-bl2.html#cb139-2" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">kruskal_test</span>(breakpoint_count <span class="sc">~</span> outlier)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 6
##   .y.                  n statistic    df     p method        
## * &lt;chr&gt;            &lt;int&gt;     &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;         
## 1 breakpoint_count  1375     0.928     1 0.335 Kruskal-Wallis</code></pre>
<div class="sourceCode" id="cb141"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb141-1"><a href="bl1-and-bl2.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="co"># removing low-quality cells</span></span>
<span id="cb141-2"><a href="bl1-and-bl2.html#cb141-2" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> bl2[,<span class="fu">colData</span>(bl2)<span class="sc">$</span>outlier <span class="sc">==</span> <span class="cn">FALSE</span>]</span></code></pre></div>
</div>
<div id="umap-and-clustering-1" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> UMAP and clustering</h3>
<div class="sourceCode" id="cb142"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb142-1"><a href="bl1-and-bl2.html#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running UMAP and storing into the copykit object</span></span>
<span id="cb142-2"><a href="bl1-and-bl2.html#cb142-2" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(bl2)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb147"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb147-1"><a href="bl1-and-bl2.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Grid Search of Jaccard Similarity (cluster stability)</span></span>
<span id="cb147-2"><a href="bl1-and-bl2.html#cb147-2" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(bl2)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 11 with median jaccard similarity of: 0.928</code></pre>
<div class="sourceCode" id="cb151"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb151-1"><a href="bl1-and-bl2.html#cb151-1" aria-hidden="true" tabindex="-1"></a>bl2_primary_suggestedK_p <span class="ot">&lt;-</span> <span class="fu">plotSuggestedK</span>(bl2)</span>
<span id="cb151-2"><a href="bl1-and-bl2.html#cb151-2" aria-hidden="true" tabindex="-1"></a>bl2_primary_suggestedK_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl2_umap_cl-1.png" width="672" /></p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb152-1"><a href="bl1-and-bl2.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding clusters using the suggested k value as per default settings, </span></span>
<span id="cb152-2"><a href="bl1-and-bl2.html#cb152-2" aria-hidden="true" tabindex="-1"></a><span class="co"># can be changed with the argument k_subclones</span></span>
<span id="cb152-3"><a href="bl1-and-bl2.html#cb152-3" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(bl2)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 11</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 62 outliers cells (group &#39;c0&#39;)</code></pre>
<pre><code>## Found 19 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb158"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb158-1"><a href="bl1-and-bl2.html#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HDBSCAN is an outlier aware clustering algorithm</span></span>
<span id="cb158-2"><a href="bl1-and-bl2.html#cb158-2" aria-hidden="true" tabindex="-1"></a><span class="co"># in this analysis all cells marked as outliers (c0) from hdbscan are excluded.</span></span>
<span id="cb158-3"><a href="bl1-and-bl2.html#cb158-3" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> bl2[,<span class="fu">colData</span>(bl2)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span>
<span id="cb158-4"><a href="bl1-and-bl2.html#cb158-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-5"><a href="bl1-and-bl2.html#cb158-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting UMAP colored by subclones</span></span>
<span id="cb158-6"><a href="bl1-and-bl2.html#cb158-6" aria-hidden="true" tabindex="-1"></a>bl2_umap_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(bl2, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: subclones.</code></pre>
<div class="sourceCode" id="cb161"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb161-1"><a href="bl1-and-bl2.html#cb161-1" aria-hidden="true" tabindex="-1"></a>bl2_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl2_umap_cl-2.png" width="672" /></p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb162-1"><a href="bl1-and-bl2.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating consensus matrix and consensus phylo, </span></span>
<span id="cb162-2"><a href="bl1-and-bl2.html#cb162-2" aria-hidden="true" tabindex="-1"></a><span class="co"># this will be used by copykit to order the subclones on the heatmap</span></span>
<span id="cb162-3"><a href="bl1-and-bl2.html#cb162-3" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(bl2)</span>
<span id="cb162-4"><a href="bl1-and-bl2.html#cb162-4" aria-hidden="true" tabindex="-1"></a>bl2 <span class="ot">&lt;-</span> <span class="fu">runConsensusPhylo</span>(bl2)</span></code></pre></div>
</div>
<div id="subclones-heatmap-1" class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Subclones Heatmap</h3>
<div class="sourceCode" id="cb163"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb163-1"><a href="bl1-and-bl2.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting copy number heatmap and annotating with subclones</span></span>
<span id="cb163-2"><a href="bl1-and-bl2.html#cb163-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl2, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/bl2_subclones-1.png" width="768" /></p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb165-1"><a href="bl1-and-bl2.html#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding macro-spatial information from file names to colData</span></span>
<span id="cb165-2"><a href="bl1-and-bl2.html#cb165-2" aria-hidden="true" tabindex="-1"></a><span class="co"># and renaming to (S)ectors</span></span>
<span id="cb165-3"><a href="bl1-and-bl2.html#cb165-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(bl2)<span class="sc">$</span>sectors <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(<span class="fu">colData</span>(bl2)<span class="sc">$</span>sample,</span>
<span id="cb165-4"><a href="bl1-and-bl2.html#cb165-4" aria-hidden="true" tabindex="-1"></a>                                                     <span class="st">&quot;L[0-9]+&quot;</span>)</span>
<span id="cb165-5"><a href="bl1-and-bl2.html#cb165-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(bl2)<span class="sc">$</span>sectors <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(<span class="fu">colData</span>(bl2)<span class="sc">$</span>sectors, </span>
<span id="cb165-6"><a href="bl1-and-bl2.html#cb165-6" aria-hidden="true" tabindex="-1"></a>                                                     <span class="st">&quot;L&quot;</span>, <span class="st">&quot;S&quot;</span>)</span>
<span id="cb165-7"><a href="bl1-and-bl2.html#cb165-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the UMAP colored by sectors</span></span>
<span id="cb165-8"><a href="bl1-and-bl2.html#cb165-8" aria-hidden="true" tabindex="-1"></a>bl2_umap_sectors_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(bl2, <span class="at">label =</span> <span class="st">&#39;sectors&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: sectors.</code></pre>
<div class="sourceCode" id="cb168"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb168-1"><a href="bl1-and-bl2.html#cb168-1" aria-hidden="true" tabindex="-1"></a>bl2_umap_sectors_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl2_spatial_sections-1.png" width="672" /></p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb169-1"><a href="bl1-and-bl2.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating pie charts plots of the proportions of clones on each sector</span></span>
<span id="cb169-2"><a href="bl1-and-bl2.html#cb169-2" aria-hidden="true" tabindex="-1"></a>bl2_focis <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">colData</span>(bl2)<span class="sc">$</span>sectors)</span>
<span id="cb169-3"><a href="bl1-and-bl2.html#cb169-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-4"><a href="bl1-and-bl2.html#cb169-4" aria-hidden="true" tabindex="-1"></a>bl2_foci_pies <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb169-5"><a href="bl1-and-bl2.html#cb169-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-6"><a href="bl1-and-bl2.html#cb169-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(bl2_focis)) {</span>
<span id="cb169-7"><a href="bl1-and-bl2.html#cb169-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-8"><a href="bl1-and-bl2.html#cb169-8" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(bl2)) <span class="sc">%&gt;%</span></span>
<span id="cb169-9"><a href="bl1-and-bl2.html#cb169-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sectors <span class="sc">==</span> bl2_focis[i])</span>
<span id="cb169-10"><a href="bl1-and-bl2.html#cb169-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-11"><a href="bl1-and-bl2.html#cb169-11" aria-hidden="true" tabindex="-1"></a>  bl2_foci_pies[[i]] <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df) <span class="sc">+</span></span>
<span id="cb169-12"><a href="bl1-and-bl2.html#cb169-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> sectors, <span class="at">fill =</span> subclones),</span>
<span id="cb169-13"><a href="bl1-and-bl2.html#cb169-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">stat =</span> <span class="st">&#39;identity&#39;</span>) <span class="sc">+</span></span>
<span id="cb169-14"><a href="bl1-and-bl2.html#cb169-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb169-15"><a href="bl1-and-bl2.html#cb169-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">subclones_pal</span>(),</span>
<span id="cb169-16"><a href="bl1-and-bl2.html#cb169-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">limits =</span> force) <span class="sc">+</span></span>
<span id="cb169-17"><a href="bl1-and-bl2.html#cb169-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_polar</span>(<span class="at">theta =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb169-18"><a href="bl1-and-bl2.html#cb169-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(bl2_focis[i])</span>
<span id="cb169-19"><a href="bl1-and-bl2.html#cb169-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-20"><a href="bl1-and-bl2.html#cb169-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb169-21"><a href="bl1-and-bl2.html#cb169-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-22"><a href="bl1-and-bl2.html#cb169-22" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(bl2_foci_pies)</span></code></pre></div>
<p><img src="_main_files/figure-html/bl2_spatial_sections-2.png" width="672" /></p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb170-1"><a href="bl1-and-bl2.html#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting up a theme for the next plot</span></span>
<span id="cb170-2"><a href="bl1-and-bl2.html#cb170-2" aria-hidden="true" tabindex="-1"></a>my_theme <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb170-3"><a href="bl1-and-bl2.html#cb170-3" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb170-4"><a href="bl1-and-bl2.html#cb170-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> <span class="st">&quot;gray28&quot;</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb170-5"><a href="bl1-and-bl2.html#cb170-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb170-6"><a href="bl1-and-bl2.html#cb170-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb170-7"><a href="bl1-and-bl2.html#cb170-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> <span class="st">&quot;gray28&quot;</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb170-8"><a href="bl1-and-bl2.html#cb170-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb170-9"><a href="bl1-and-bl2.html#cb170-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb170-10"><a href="bl1-and-bl2.html#cb170-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb170-11"><a href="bl1-and-bl2.html#cb170-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb170-12"><a href="bl1-and-bl2.html#cb170-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>)</span>
<span id="cb170-13"><a href="bl1-and-bl2.html#cb170-13" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb170-14"><a href="bl1-and-bl2.html#cb170-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;umap1&quot;</span>),</span>
<span id="cb170-15"><a href="bl1-and-bl2.html#cb170-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;umap2&quot;</span>)</span>
<span id="cb170-16"><a href="bl1-and-bl2.html#cb170-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb170-17"><a href="bl1-and-bl2.html#cb170-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-18"><a href="bl1-and-bl2.html#cb170-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the UMAP colored by sector and using ggrepel to label the subclones</span></span>
<span id="cb170-19"><a href="bl1-and-bl2.html#cb170-19" aria-hidden="true" tabindex="-1"></a>bl2_lfoci_umap <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">reducedDim</span>(bl2, <span class="st">&#39;umap&#39;</span>))</span>
<span id="cb170-20"><a href="bl1-and-bl2.html#cb170-20" aria-hidden="true" tabindex="-1"></a>bl2_lfoci_umap<span class="sc">$</span>sectors <span class="ot">&lt;-</span> <span class="fu">colData</span>(bl2)<span class="sc">$</span>sectors</span>
<span id="cb170-21"><a href="bl1-and-bl2.html#cb170-21" aria-hidden="true" tabindex="-1"></a>bl2_lfoci_umap<span class="sc">$</span>subclones <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">colData</span>(bl2)<span class="sc">$</span>subclones)</span>
<span id="cb170-22"><a href="bl1-and-bl2.html#cb170-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Avoiding overplotting of text</span></span>
<span id="cb170-23"><a href="bl1-and-bl2.html#cb170-23" aria-hidden="true" tabindex="-1"></a>bl2_lfoci_umap<span class="sc">$</span>subclones <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="sc">!</span><span class="fu">duplicated</span>(bl2_lfoci_umap<span class="sc">$</span>subclones),</span>
<span id="cb170-24"><a href="bl1-and-bl2.html#cb170-24" aria-hidden="true" tabindex="-1"></a>                                   bl2_lfoci_umap<span class="sc">$</span>subclones, <span class="st">&quot;&quot;</span>)</span>
<span id="cb170-25"><a href="bl1-and-bl2.html#cb170-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-26"><a href="bl1-and-bl2.html#cb170-26" aria-hidden="true" tabindex="-1"></a>bl2_lfoci_umap_p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(bl2_lfoci_umap, <span class="fu">aes</span>(V1, V2, <span class="at">label =</span> subclones)) <span class="sc">+</span></span>
<span id="cb170-27"><a href="bl1-and-bl2.html#cb170-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill =</span> sectors), <span class="at">shape =</span> <span class="dv">21</span>,</span>
<span id="cb170-28"><a href="bl1-and-bl2.html#cb170-28" aria-hidden="true" tabindex="-1"></a>             <span class="at">stroke =</span> <span class="fl">0.1</span>,</span>
<span id="cb170-29"><a href="bl1-and-bl2.html#cb170-29" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb170-30"><a href="bl1-and-bl2.html#cb170-30" aria-hidden="true" tabindex="-1"></a>  ggnewscale<span class="sc">::</span><span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb170-31"><a href="bl1-and-bl2.html#cb170-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">color =</span> subclones),</span>
<span id="cb170-32"><a href="bl1-and-bl2.html#cb170-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">min.segment.length =</span> <span class="dv">0</span>,</span>
<span id="cb170-33"><a href="bl1-and-bl2.html#cb170-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">box.padding =</span> <span class="fl">0.01</span>,</span>
<span id="cb170-34"><a href="bl1-and-bl2.html#cb170-34" aria-hidden="true" tabindex="-1"></a>                   <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb170-35"><a href="bl1-and-bl2.html#cb170-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb170-36"><a href="bl1-and-bl2.html#cb170-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">na.rm =</span> T) <span class="sc">+</span></span>
<span id="cb170-37"><a href="bl1-and-bl2.html#cb170-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">subclones_pal</span>(),</span>
<span id="cb170-38"><a href="bl1-and-bl2.html#cb170-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">limits =</span> force) <span class="sc">+</span></span>
<span id="cb170-39"><a href="bl1-and-bl2.html#cb170-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb170-40"><a href="bl1-and-bl2.html#cb170-40" aria-hidden="true" tabindex="-1"></a>  my_theme</span>
<span id="cb170-41"><a href="bl1-and-bl2.html#cb170-41" aria-hidden="true" tabindex="-1"></a>bl2_lfoci_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl2_spatial_sections-3.png" width="672" /></p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb171-1"><a href="bl1-and-bl2.html#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting consensus heatmap with a frequency annotation of sectors </span></span>
<span id="cb171-2"><a href="bl1-and-bl2.html#cb171-2" aria-hidden="true" tabindex="-1"></a><span class="co"># by subclone and annotating genes of interest</span></span>
<span id="cb171-3"><a href="bl1-and-bl2.html#cb171-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(bl2, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">consensus =</span> T, <span class="at">group =</span> <span class="st">&#39;sectors&#39;</span>,</span>
<span id="cb171-4"><a href="bl1-and-bl2.html#cb171-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">genes =</span> <span class="fu">c</span>(</span>
<span id="cb171-5"><a href="bl1-and-bl2.html#cb171-5" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb171-6"><a href="bl1-and-bl2.html#cb171-6" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb171-7"><a href="bl1-and-bl2.html#cb171-7" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;GATA1&quot;</span>,</span>
<span id="cb171-8"><a href="bl1-and-bl2.html#cb171-8" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;SOX2&quot;</span>,</span>
<span id="cb171-9"><a href="bl1-and-bl2.html#cb171-9" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;ERBB2&quot;</span>,</span>
<span id="cb171-10"><a href="bl1-and-bl2.html#cb171-10" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb171-11"><a href="bl1-and-bl2.html#cb171-11" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;RB1&quot;</span>,</span>
<span id="cb171-12"><a href="bl1-and-bl2.html#cb171-12" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb171-13"><a href="bl1-and-bl2.html#cb171-13" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;MMP3&quot;</span>,</span>
<span id="cb171-14"><a href="bl1-and-bl2.html#cb171-14" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;BRCA1&quot;</span>,</span>
<span id="cb171-15"><a href="bl1-and-bl2.html#cb171-15" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;BRCA2&quot;</span>,</span>
<span id="cb171-16"><a href="bl1-and-bl2.html#cb171-16" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;FHIT&quot;</span></span>
<span id="cb171-17"><a href="bl1-and-bl2.html#cb171-17" aria-hidden="true" tabindex="-1"></a>            ))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/bl2_spatial_sections-4.png" width="672" /></p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb173-1"><a href="bl1-and-bl2.html#cb173-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating Diversity Indexes by sector</span></span>
<span id="cb173-2"><a href="bl1-and-bl2.html#cb173-2" aria-hidden="true" tabindex="-1"></a>bl2_meta <span class="ot">&lt;-</span> <span class="fu">colData</span>(bl2)</span>
<span id="cb173-3"><a href="bl1-and-bl2.html#cb173-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-4"><a href="bl1-and-bl2.html#cb173-4" aria-hidden="true" tabindex="-1"></a>bl2_meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(bl2_meta)</span>
<span id="cb173-5"><a href="bl1-and-bl2.html#cb173-5" aria-hidden="true" tabindex="-1"></a>bl2_meta <span class="ot">&lt;-</span> bl2_meta[<span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>, <span class="st">&#39;sectors&#39;</span>)]</span>
<span id="cb173-6"><a href="bl1-and-bl2.html#cb173-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-7"><a href="bl1-and-bl2.html#cb173-7" aria-hidden="true" tabindex="-1"></a>bl2_L1_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl2_meta, <span class="st">&quot;S1&quot;</span>)</span>
<span id="cb173-8"><a href="bl1-and-bl2.html#cb173-8" aria-hidden="true" tabindex="-1"></a>bl2_L4_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl2_meta, <span class="st">&quot;S4&quot;</span>)</span>
<span id="cb173-9"><a href="bl1-and-bl2.html#cb173-9" aria-hidden="true" tabindex="-1"></a>bl2_L5_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl2_meta, <span class="st">&quot;S5&quot;</span>)</span>
<span id="cb173-10"><a href="bl1-and-bl2.html#cb173-10" aria-hidden="true" tabindex="-1"></a>bl2_L6_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl2_meta, <span class="st">&quot;S6&quot;</span>)</span>
<span id="cb173-11"><a href="bl1-and-bl2.html#cb173-11" aria-hidden="true" tabindex="-1"></a>bl2_L7_div <span class="ot">&lt;-</span> <span class="fu">shan_sample</span>(bl2_meta, <span class="st">&quot;S7&quot;</span>)</span>
<span id="cb173-12"><a href="bl1-and-bl2.html#cb173-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-13"><a href="bl1-and-bl2.html#cb173-13" aria-hidden="true" tabindex="-1"></a>bl2_div <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb173-14"><a href="bl1-and-bl2.html#cb173-14" aria-hidden="true" tabindex="-1"></a>  bl2_L1_div,</span>
<span id="cb173-15"><a href="bl1-and-bl2.html#cb173-15" aria-hidden="true" tabindex="-1"></a>  bl2_L4_div,</span>
<span id="cb173-16"><a href="bl1-and-bl2.html#cb173-16" aria-hidden="true" tabindex="-1"></a>  bl2_L5_div,</span>
<span id="cb173-17"><a href="bl1-and-bl2.html#cb173-17" aria-hidden="true" tabindex="-1"></a>  bl2_L6_div,</span>
<span id="cb173-18"><a href="bl1-and-bl2.html#cb173-18" aria-hidden="true" tabindex="-1"></a>  bl2_L7_div</span>
<span id="cb173-19"><a href="bl1-and-bl2.html#cb173-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb173-20"><a href="bl1-and-bl2.html#cb173-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-21"><a href="bl1-and-bl2.html#cb173-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-22"><a href="bl1-and-bl2.html#cb173-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting Shannon Diversity Index</span></span>
<span id="cb173-23"><a href="bl1-and-bl2.html#cb173-23" aria-hidden="true" tabindex="-1"></a>bl2_div_p <span class="ot">&lt;-</span> bl2_div <span class="sc">%&gt;%</span></span>
<span id="cb173-24"><a href="bl1-and-bl2.html#cb173-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb173-25"><a href="bl1-and-bl2.html#cb173-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(</span>
<span id="cb173-26"><a href="bl1-and-bl2.html#cb173-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">fct_reorder</span>(foci, div),</span>
<span id="cb173-27"><a href="bl1-and-bl2.html#cb173-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymin =</span> lci,</span>
<span id="cb173-28"><a href="bl1-and-bl2.html#cb173-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymax =</span> uci</span>
<span id="cb173-29"><a href="bl1-and-bl2.html#cb173-29" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb173-30"><a href="bl1-and-bl2.html#cb173-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> .<span class="dv">1</span>,</span>
<span id="cb173-31"><a href="bl1-and-bl2.html#cb173-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb173-32"><a href="bl1-and-bl2.html#cb173-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb173-33"><a href="bl1-and-bl2.html#cb173-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">fct_reorder</span>(foci, div),</span>
<span id="cb173-34"><a href="bl1-and-bl2.html#cb173-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> div,</span>
<span id="cb173-35"><a href="bl1-and-bl2.html#cb173-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> foci</span>
<span id="cb173-36"><a href="bl1-and-bl2.html#cb173-36" aria-hidden="true" tabindex="-1"></a>  ), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb173-37"><a href="bl1-and-bl2.html#cb173-37" aria-hidden="true" tabindex="-1"></a>  cowplot<span class="sc">::</span><span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb173-38"><a href="bl1-and-bl2.html#cb173-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb173-39"><a href="bl1-and-bl2.html#cb173-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(</span>
<span id="cb173-40"><a href="bl1-and-bl2.html#cb173-40" aria-hidden="true" tabindex="-1"></a>          <span class="at">angle =</span> <span class="dv">90</span>,</span>
<span id="cb173-41"><a href="bl1-and-bl2.html#cb173-41" aria-hidden="true" tabindex="-1"></a>          <span class="at">vjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb173-42"><a href="bl1-and-bl2.html#cb173-42" aria-hidden="true" tabindex="-1"></a>          <span class="at">hjust =</span> <span class="dv">1</span></span>
<span id="cb173-43"><a href="bl1-and-bl2.html#cb173-43" aria-hidden="true" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb173-44"><a href="bl1-and-bl2.html#cb173-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> scales<span class="sc">::</span><span class="fu">pretty_breaks</span>(<span class="at">n =</span> <span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb173-45"><a href="bl1-and-bl2.html#cb173-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;shannon diversity index&quot;</span>) <span class="sc">+</span></span>
<span id="cb173-46"><a href="bl1-and-bl2.html#cb173-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb173-47"><a href="bl1-and-bl2.html#cb173-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb173-48"><a href="bl1-and-bl2.html#cb173-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-49"><a href="bl1-and-bl2.html#cb173-49" aria-hidden="true" tabindex="-1"></a>bl2_div_p</span></code></pre></div>
<p><img src="_main_files/figure-html/bl2_spatial_sections-5.png" width="672" /></p>

<p>The following code analysis the breast metastatic sample with metastasis to the liver and to the pleural effusion.
This works as an example on how to employ CopyKit for the analysis of multiple datasets.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bm1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/navinlabcode/copykit/blob/master/02-bl1_bl2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
