<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 CM1 and CM2 | _main.knit</title>
  <meta name="description" content="<p>This is the code from the figures presented in the CopyKit manuscript -
Resolving clonal substructure from single cell genomic data using CopyKit.</p>" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="4 CM1 and CM2 | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is the code from the figures presented in the CopyKit manuscript -
Resolving clonal substructure from single cell genomic data using CopyKit.</p>" />
  <meta name="github-repo" content="navinlabcode/copykit" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 CM1 and CM2 | _main.knit" />
  
  <meta name="twitter:description" content="<p>This is the code from the figures presented in the CopyKit manuscript -
Resolving clonal substructure from single cell genomic data using CopyKit.</p>" />
  

<meta name="author" content="Darlan Conterno Minussi" />


<meta name="date" content="2022-02-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bm1.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CopyKit Manuscript Code</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html"><i class="fa fa-check"></i><b>2</b> BL1 and BL2</a>
<ul>
<li class="chapter" data-level="2.1" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#bl1"><i class="fa fa-check"></i><b>2.1</b> BL1</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#filtering"><i class="fa fa-check"></i><b>2.1.1</b> Filtering</a></li>
<li class="chapter" data-level="2.1.2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#umap-and-clustering"><i class="fa fa-check"></i><b>2.1.2</b> UMAP and clustering</a></li>
<li class="chapter" data-level="2.1.3" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#subclones-heatmap"><i class="fa fa-check"></i><b>2.1.3</b> Subclones Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#bl2"><i class="fa fa-check"></i><b>2.2</b> BL2</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#filtering-1"><i class="fa fa-check"></i><b>2.2.1</b> Filtering</a></li>
<li class="chapter" data-level="2.2.2" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#umap-and-clustering-1"><i class="fa fa-check"></i><b>2.2.2</b> UMAP and clustering</a></li>
<li class="chapter" data-level="2.2.3" data-path="bl1-and-bl2.html"><a href="bl1-and-bl2.html#subclones-heatmap-1"><i class="fa fa-check"></i><b>2.2.3</b> Subclones Heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="bm1.html"><a href="bm1.html"><i class="fa fa-check"></i><b>3</b> BM1</a>
<ul>
<li class="chapter" data-level="3.1" data-path="bm1.html"><a href="bm1.html#reading-datasets"><i class="fa fa-check"></i><b>3.1</b> Reading datasets</a></li>
<li class="chapter" data-level="3.2" data-path="bm1.html"><a href="bm1.html#running-umap-and-clustering"><i class="fa fa-check"></i><b>3.2</b> Running UMAP and Clustering</a></li>
<li class="chapter" data-level="3.3" data-path="bm1.html"><a href="bm1.html#consensus-tree"><i class="fa fa-check"></i><b>3.3</b> Consensus tree</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html"><i class="fa fa-check"></i><b>4</b> CM1 and CM2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1"><i class="fa fa-check"></i><b>4.1</b> CM1</a></li>
<li class="chapter" data-level="4.2" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1-colon"><i class="fa fa-check"></i><b>4.2</b> CM1 colon</a></li>
<li class="chapter" data-level="4.3" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1-liver"><i class="fa fa-check"></i><b>4.3</b> CM1 liver</a></li>
<li class="chapter" data-level="4.4" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm1-merging"><i class="fa fa-check"></i><b>4.4</b> CM1 merging</a></li>
<li class="chapter" data-level="4.5" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2"><i class="fa fa-check"></i><b>4.5</b> CM2</a></li>
<li class="chapter" data-level="4.6" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2-colon"><i class="fa fa-check"></i><b>4.6</b> CM2 colon</a></li>
<li class="chapter" data-level="4.7" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2-liver"><i class="fa fa-check"></i><b>4.7</b> CM2 liver</a></li>
<li class="chapter" data-level="4.8" data-path="cm1-and-cm2.html"><a href="cm1-and-cm2.html#cm2-merging"><i class="fa fa-check"></i><b>4.8</b> CM2 merging</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>5</b> Session Info</a></li>
<li class="chapter" data-level="6" data-path="copykit-paper.html"><a href="copykit-paper.html"><i class="fa fa-check"></i><b>6</b> CopyKit Paper</a>
<ul>
<li class="chapter" data-level="6.1" data-path="copykit-paper.html"><a href="copykit-paper.html#manuscript-code"><i class="fa fa-check"></i><b>6.1</b> Manuscript Code</a></li>
<li class="chapter" data-level="6.2" data-path="copykit-paper.html"><a href="copykit-paper.html#copykit-installation"><i class="fa fa-check"></i><b>6.2</b> CopyKit installation</a></li>
<li class="chapter" data-level="6.3" data-path="copykit-paper.html"><a href="copykit-paper.html#documentation"><i class="fa fa-check"></i><b>6.3</b> Documentation</a></li>
<li class="chapter" data-level="6.4" data-path="copykit-paper.html"><a href="copykit-paper.html#datasets"><i class="fa fa-check"></i><b>6.4</b> Datasets</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cm1-and-cm2" class="section level1" number="4">
<h1><span class="header-section-number">4</span> CM1 and CM2</h1>
<div id="cm1" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> CM1</h2>
</div>
<div id="cm1-colon" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> CM1 colon</h2>
<div class="sourceCode" id="cb285"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb285-1"><a href="cm1-and-cm2.html#cb285-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running data with CopyKit for the primary colon sample</span></span>
<span id="cb285-2"><a href="cm1-and-cm2.html#cb285-2" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">runVarbin</span>(<span class="st">&quot;/mnt/lab/users/dminussi/projects/CopyKit_Manuscript_Code/datasets/CM1/colon/marked_bams&quot;</span>, <span class="at">remove_Y =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Counting reads for genome hg38 and resolution: 220kb</code></pre>
<pre><code>## 441 bam files had less than 10 mean bincounts and were removed.</code></pre>
<pre><code>## Performing GC correction.</code></pre>
<pre><code>## Smoothing bin counts.</code></pre>
<pre><code>## Running segmentation algorithm: CBS for genome hg38</code></pre>
<pre><code>## Merging levels.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb293"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb293-1"><a href="cm1-and-cm2.html#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding diploid and low-quality cells and excluding it from the copykit object</span></span>
<span id="cb293-2"><a href="cm1-and-cm2.html#cb293-2" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findOutliers</span>(cm1_tumor_primary)</span></code></pre></div>
<pre><code>## Calculating correlation matrix.</code></pre>
<pre><code>## Marked 347 cells as outliers.</code></pre>
<pre><code>## Adding information to metadata. Access with colData(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb298"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb298-1"><a href="cm1-and-cm2.html#cb298-1" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findAneuploidCells</span>(cm1_tumor_primary)</span></code></pre></div>
<pre><code>## number of iterations= 10</code></pre>
<pre><code>## Copykit detected 35 that are possibly diploid cells using a resolution of: 0.048</code></pre>
<pre><code>## Added information to colData(CopyKit).</code></pre>
<div class="sourceCode" id="cb302"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb302-1"><a href="cm1-and-cm2.html#cb302-1" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> cm1_tumor_primary[,<span class="fu">colData</span>(cm1_tumor_primary)<span class="sc">$</span>outlier <span class="sc">==</span> <span class="cn">FALSE</span>]</span>
<span id="cb302-2"><a href="cm1-and-cm2.html#cb302-2" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> cm1_tumor_primary[,<span class="fu">colData</span>(cm1_tumor_primary)<span class="sc">$</span>is_aneuploid <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb302-3"><a href="cm1-and-cm2.html#cb302-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-4"><a href="cm1-and-cm2.html#cb302-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding the tissue information to colData</span></span>
<span id="cb302-5"><a href="cm1-and-cm2.html#cb302-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm1_tumor_primary)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span> <span class="st">&#39;primary&#39;</span></span></code></pre></div>
</div>
<div id="cm1-liver" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> CM1 liver</h2>
<div class="sourceCode" id="cb303"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb303-1"><a href="cm1-and-cm2.html#cb303-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running data with CopyKit for the liver sample</span></span>
<span id="cb303-2"><a href="cm1-and-cm2.html#cb303-2" aria-hidden="true" tabindex="-1"></a>cm1_tumor_met <span class="ot">&lt;-</span></span>
<span id="cb303-3"><a href="cm1-and-cm2.html#cb303-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">runVarbin</span>(</span>
<span id="cb303-4"><a href="cm1-and-cm2.html#cb303-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;/mnt/lab/users/dminussi/projects/CopyKit_Manuscript_Code/datasets/CM1/liver/marked_bams&quot;</span>,</span>
<span id="cb303-5"><a href="cm1-and-cm2.html#cb303-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">remove_Y =</span> <span class="cn">TRUE</span></span>
<span id="cb303-6"><a href="cm1-and-cm2.html#cb303-6" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## Counting reads for genome hg38 and resolution: 220kb</code></pre>
<pre><code>## 549 bam files had less than 10 mean bincounts and were removed.</code></pre>
<pre><code>## Performing GC correction.</code></pre>
<pre><code>## Smoothing bin counts.</code></pre>
<pre><code>## Running segmentation algorithm: CBS for genome hg38</code></pre>
<pre><code>## Merging levels.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb311"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb311-1"><a href="cm1-and-cm2.html#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding diploid and low-quality cells and excluding it from the copykit object</span></span>
<span id="cb311-2"><a href="cm1-and-cm2.html#cb311-2" aria-hidden="true" tabindex="-1"></a>cm1_tumor_met <span class="ot">&lt;-</span> <span class="fu">findOutliers</span>(cm1_tumor_met)</span></code></pre></div>
<pre><code>## Calculating correlation matrix.</code></pre>
<pre><code>## Marked 85 cells as outliers.</code></pre>
<pre><code>## Adding information to metadata. Access with colData(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb316"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb316-1"><a href="cm1-and-cm2.html#cb316-1" aria-hidden="true" tabindex="-1"></a>cm1_tumor_met <span class="ot">&lt;-</span> <span class="fu">findAneuploidCells</span>(cm1_tumor_met)</span></code></pre></div>
<pre><code>## number of iterations= 8</code></pre>
<pre><code>## Copykit detected 13 that are possibly diploid cells using a resolution of: 0.053</code></pre>
<pre><code>## Added information to colData(CopyKit).</code></pre>
<div class="sourceCode" id="cb320"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb320-1"><a href="cm1-and-cm2.html#cb320-1" aria-hidden="true" tabindex="-1"></a>cm1_tumor_met <span class="ot">&lt;-</span></span>
<span id="cb320-2"><a href="cm1-and-cm2.html#cb320-2" aria-hidden="true" tabindex="-1"></a>  cm1_tumor_met[, <span class="fu">colData</span>(cm1_tumor_met)<span class="sc">$</span>outlier <span class="sc">==</span> <span class="cn">FALSE</span>]</span>
<span id="cb320-3"><a href="cm1-and-cm2.html#cb320-3" aria-hidden="true" tabindex="-1"></a>cm1_tumor_met <span class="ot">&lt;-</span></span>
<span id="cb320-4"><a href="cm1-and-cm2.html#cb320-4" aria-hidden="true" tabindex="-1"></a>  cm1_tumor_met[, <span class="fu">colData</span>(cm1_tumor_met)<span class="sc">$</span>is_aneuploid <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb320-5"><a href="cm1-and-cm2.html#cb320-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb320-6"><a href="cm1-and-cm2.html#cb320-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding the tissue information to colData</span></span>
<span id="cb320-7"><a href="cm1-and-cm2.html#cb320-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm1_tumor_met)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span> <span class="st">&#39;metastasis&#39;</span></span></code></pre></div>
</div>
<div id="cm1-merging" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> CM1 merging</h2>
<div class="sourceCode" id="cb321"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb321-1"><a href="cm1-and-cm2.html#cb321-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merging the three copykit objects</span></span>
<span id="cb321-2"><a href="cm1-and-cm2.html#cb321-2" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> <span class="fu">cbind</span>(cm1_tumor_primary,</span>
<span id="cb321-3"><a href="cm1-and-cm2.html#cb321-3" aria-hidden="true" tabindex="-1"></a>                    cm1_tumor_met)</span>
<span id="cb321-4"><a href="cm1-and-cm2.html#cb321-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb321-5"><a href="cm1-and-cm2.html#cb321-5" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(cm1_merged)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb326"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb326-1"><a href="cm1-and-cm2.html#cb326-1" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(cm1_merged)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 44 with median jaccard similarity of: 0.97</code></pre>
<div class="sourceCode" id="cb330"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb330-1"><a href="cm1-and-cm2.html#cb330-1" aria-hidden="true" tabindex="-1"></a>cm1_merged_suggestedk <span class="ot">&lt;-</span> <span class="fu">plotSuggestedK</span>(cm1_merged)</span>
<span id="cb330-2"><a href="cm1-and-cm2.html#cb330-2" aria-hidden="true" tabindex="-1"></a>cm1_merged_suggestedk</span></code></pre></div>
<p><img src="_main_files/figure-html/cm1_merge-1.png" width="672" /></p>
<div class="sourceCode" id="cb331"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb331-1"><a href="cm1-and-cm2.html#cb331-1" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(cm1_merged)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 44</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 226 outliers cells (group &#39;c0&#39;)</code></pre>
<pre><code>## Found 22 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb337"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb337-1"><a href="cm1-and-cm2.html#cb337-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HDBSCAN is an outlier aware clustering algorithm</span></span>
<span id="cb337-2"><a href="cm1-and-cm2.html#cb337-2" aria-hidden="true" tabindex="-1"></a><span class="co"># in this analysis all cells marked as outliers (c0) from hdbscan are excluded.</span></span>
<span id="cb337-3"><a href="cm1-and-cm2.html#cb337-3" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> cm1_merged[, <span class="fu">colData</span>(cm1_merged)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span>
<span id="cb337-4"><a href="cm1-and-cm2.html#cb337-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb337-5"><a href="cm1-and-cm2.html#cb337-5" aria-hidden="true" tabindex="-1"></a>cm1_merged_tp_umap_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(cm1_merged, <span class="at">label =</span> <span class="st">&#39;timepoint&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: timepoint.</code></pre>
<div class="sourceCode" id="cb340"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb340-1"><a href="cm1-and-cm2.html#cb340-1" aria-hidden="true" tabindex="-1"></a>cm1_merged_tp_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/cm1_merge-2.png" width="672" /></p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb341-1"><a href="cm1-and-cm2.html#cb341-1" aria-hidden="true" tabindex="-1"></a>cm1_merged_umap_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(cm1_merged, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: subclones.</code></pre>
<div class="sourceCode" id="cb344"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb344-1"><a href="cm1-and-cm2.html#cb344-1" aria-hidden="true" tabindex="-1"></a>cm1_merged_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/cm1_merge-3.png" width="672" /></p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb345-1"><a href="cm1-and-cm2.html#cb345-1" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(cm1_merged)</span>
<span id="cb345-2"><a href="cm1-and-cm2.html#cb345-2" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> <span class="fu">runConsensusPhylo</span>(cm1_merged)</span>
<span id="cb345-3"><a href="cm1-and-cm2.html#cb345-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-4"><a href="cm1-and-cm2.html#cb345-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm1_merged)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span></span>
<span id="cb345-5"><a href="cm1-and-cm2.html#cb345-5" aria-hidden="true" tabindex="-1"></a>  forcats<span class="sc">::</span><span class="fu">fct_relevel</span>(<span class="fu">colData</span>(cm1_merged)<span class="sc">$</span>timepoint,</span>
<span id="cb345-6"><a href="cm1-and-cm2.html#cb345-6" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">c</span>(<span class="st">&quot;primary&quot;</span>, <span class="st">&quot;metastasis&quot;</span>))</span>
<span id="cb345-7"><a href="cm1-and-cm2.html#cb345-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-8"><a href="cm1-and-cm2.html#cb345-8" aria-hidden="true" tabindex="-1"></a>cm1_selected_genes <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb345-9"><a href="cm1-and-cm2.html#cb345-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;SMAD3&quot;</span>,</span>
<span id="cb345-10"><a href="cm1-and-cm2.html#cb345-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;FHIT&quot;</span>,</span>
<span id="cb345-11"><a href="cm1-and-cm2.html#cb345-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;APC&quot;</span>,</span>
<span id="cb345-12"><a href="cm1-and-cm2.html#cb345-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;SOX4&quot;</span>,</span>
<span id="cb345-13"><a href="cm1-and-cm2.html#cb345-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;IGFBP7&quot;</span>,</span>
<span id="cb345-14"><a href="cm1-and-cm2.html#cb345-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;CDK8&quot;</span>,</span>
<span id="cb345-15"><a href="cm1-and-cm2.html#cb345-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb345-16"><a href="cm1-and-cm2.html#cb345-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb345-17"><a href="cm1-and-cm2.html#cb345-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&#39;TP53&#39;</span>,</span>
<span id="cb345-18"><a href="cm1-and-cm2.html#cb345-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;GATA4&quot;</span>,</span>
<span id="cb345-19"><a href="cm1-and-cm2.html#cb345-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;CHEK1&quot;</span>,</span>
<span id="cb345-20"><a href="cm1-and-cm2.html#cb345-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;TGFB1&quot;</span>,</span>
<span id="cb345-21"><a href="cm1-and-cm2.html#cb345-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;TIAM1&quot;</span></span>
<span id="cb345-22"><a href="cm1-and-cm2.html#cb345-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb345-23"><a href="cm1-and-cm2.html#cb345-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-24"><a href="cm1-and-cm2.html#cb345-24" aria-hidden="true" tabindex="-1"></a>cm1_merged_selected_hvg_gc <span class="ot">&lt;-</span></span>
<span id="cb345-25"><a href="cm1-and-cm2.html#cb345-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotGeneCopy</span>(cm1_merged,</span>
<span id="cb345-26"><a href="cm1-and-cm2.html#cb345-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">genes =</span> cm1_selected_genes,</span>
<span id="cb345-27"><a href="cm1-and-cm2.html#cb345-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">label =</span> <span class="st">&#39;timepoint&#39;</span>,</span>
<span id="cb345-28"><a href="cm1-and-cm2.html#cb345-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">dodge.width =</span> .<span class="dv">8</span>) <span class="sc">+</span> <span class="fu">scale_fill_hue</span>(<span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Coloring by: timepoint</code></pre>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which will replace the existing
## scale.</code></pre>
<div class="sourceCode" id="cb348"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb348-1"><a href="cm1-and-cm2.html#cb348-1" aria-hidden="true" tabindex="-1"></a>cm1_merged_selected_hvg_gc</span></code></pre></div>
<p><img src="_main_files/figure-html/cm1_merge-4.png" width="672" /></p>
<p>To root the tree, we will use an inferred Most Recent Common Ancestral from the primary tumor and provide that as an argument to the runConsensusPhylo function.
This consensus tree will be used by plotHeatmap to order the subclones on the plot</p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb349-1"><a href="cm1-and-cm2.html#cb349-1" aria-hidden="true" tabindex="-1"></a><span class="co"># clustering and inferring the MRCA from the clusters</span></span>
<span id="cb349-2"><a href="cm1-and-cm2.html#cb349-2" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(cm1_tumor_primary)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb354"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb354-1"><a href="cm1-and-cm2.html#cb354-1" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(cm1_tumor_primary)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 37 with median jaccard similarity of: 1</code></pre>
<div class="sourceCode" id="cb358"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb358-1"><a href="cm1-and-cm2.html#cb358-1" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(cm1_tumor_primary)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 37</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 54 outliers cells (group &#39;c0&#39;)</code></pre>
<pre><code>## Found 10 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb364"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb364-1"><a href="cm1-and-cm2.html#cb364-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HDBSCAN is an outlier aware clustering algorithm</span></span>
<span id="cb364-2"><a href="cm1-and-cm2.html#cb364-2" aria-hidden="true" tabindex="-1"></a><span class="co"># in this analysis all cells marked as outliers (c0) from hdbscan are excluded.</span></span>
<span id="cb364-3"><a href="cm1-and-cm2.html#cb364-3" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span></span>
<span id="cb364-4"><a href="cm1-and-cm2.html#cb364-4" aria-hidden="true" tabindex="-1"></a>  cm1_tumor_primary[, <span class="fu">colData</span>(cm1_tumor_primary)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span>
<span id="cb364-5"><a href="cm1-and-cm2.html#cb364-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb364-6"><a href="cm1-and-cm2.html#cb364-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb364-7"><a href="cm1-and-cm2.html#cb364-7" aria-hidden="true" tabindex="-1"></a><span class="co"># calculating the consensus of the Merged dataset and using the inferred</span></span>
<span id="cb364-8"><a href="cm1-and-cm2.html#cb364-8" aria-hidden="true" tabindex="-1"></a><span class="co"># primary MRCA as the root of the tree</span></span>
<span id="cb364-9"><a href="cm1-and-cm2.html#cb364-9" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(cm1_tumor_primary)</span>
<span id="cb364-10"><a href="cm1-and-cm2.html#cb364-10" aria-hidden="true" tabindex="-1"></a>cm1_tumor_primary <span class="ot">&lt;-</span> <span class="fu">inferMrca</span>(cm1_tumor_primary)</span>
<span id="cb364-11"><a href="cm1-and-cm2.html#cb364-11" aria-hidden="true" tabindex="-1"></a>cm1_merged <span class="ot">&lt;-</span> <span class="fu">runConsensusPhylo</span>(</span>
<span id="cb364-12"><a href="cm1-and-cm2.html#cb364-12" aria-hidden="true" tabindex="-1"></a>  cm1_merged,</span>
<span id="cb364-13"><a href="cm1-and-cm2.html#cb364-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">root =</span> <span class="st">&#39;user&#39;</span>,</span>
<span id="cb364-14"><a href="cm1-and-cm2.html#cb364-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">root_user =</span> <span class="fu">metadata</span>(cm1_tumor_primary)<span class="sc">$</span>inferred_mrca</span>
<span id="cb364-15"><a href="cm1-and-cm2.html#cb364-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb364-16"><a href="cm1-and-cm2.html#cb364-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb364-17"><a href="cm1-and-cm2.html#cb364-17" aria-hidden="true" tabindex="-1"></a><span class="co"># relevel factors to plot in the desired order</span></span>
<span id="cb364-18"><a href="cm1-and-cm2.html#cb364-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm1_merged)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span></span>
<span id="cb364-19"><a href="cm1-and-cm2.html#cb364-19" aria-hidden="true" tabindex="-1"></a>  forcats<span class="sc">::</span><span class="fu">fct_relevel</span>(<span class="fu">colData</span>(cm1_merged)<span class="sc">$</span>timepoint,</span>
<span id="cb364-20"><a href="cm1-and-cm2.html#cb364-20" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">c</span>(<span class="st">&quot;metastasis&quot;</span>, <span class="st">&quot;primary&quot;</span>))</span>
<span id="cb364-21"><a href="cm1-and-cm2.html#cb364-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb364-22"><a href="cm1-and-cm2.html#cb364-22" aria-hidden="true" tabindex="-1"></a><span class="co"># plotting the phylogeny with subclones labels and pie charts</span></span>
<span id="cb364-23"><a href="cm1-and-cm2.html#cb364-23" aria-hidden="true" tabindex="-1"></a><span class="co">#indicating the frequency of each timepoint</span></span>
<span id="cb364-24"><a href="cm1-and-cm2.html#cb364-24" aria-hidden="true" tabindex="-1"></a>cm1_merged_consensus_phylo <span class="ot">&lt;-</span></span>
<span id="cb364-25"><a href="cm1-and-cm2.html#cb364-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotPhylo</span>(cm1_merged,</span>
<span id="cb364-26"><a href="cm1-and-cm2.html#cb364-26" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>,</span>
<span id="cb364-27"><a href="cm1-and-cm2.html#cb364-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">consensus =</span> <span class="cn">TRUE</span>,</span>
<span id="cb364-28"><a href="cm1-and-cm2.html#cb364-28" aria-hidden="true" tabindex="-1"></a>            <span class="at">group =</span> <span class="st">&#39;timepoint&#39;</span>)</span>
<span id="cb364-29"><a href="cm1-and-cm2.html#cb364-29" aria-hidden="true" tabindex="-1"></a>cm1_merged_consensus_phylo</span></code></pre></div>
<p><img src="_main_files/figure-html/cm1_tree-1.png" width="672" /></p>
<div class="sourceCode" id="cb365"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb365-1"><a href="cm1-and-cm2.html#cb365-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(cm1_merged, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>, <span class="st">&#39;timepoint&#39;</span>))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<pre><code>## The automatically generated colors map from the minus and plus 99^th of the absolute values in
## the matrix. There are outliers in the matrix whose patterns might be hidden by this color
## mapping. You can manually set the color to `col` argument.
## 
## Use `suppressMessages()` to turn off this message.</code></pre>
<p><img src="_main_files/figure-html/cm1_heatmap-1.png" width="768" /></p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb368-1"><a href="cm1-and-cm2.html#cb368-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb368-2"><a href="cm1-and-cm2.html#cb368-2" aria-hidden="true" tabindex="-1"></a>  cm1_merged,</span>
<span id="cb368-3"><a href="cm1-and-cm2.html#cb368-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>),</span>
<span id="cb368-4"><a href="cm1-and-cm2.html#cb368-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">consensus =</span> <span class="cn">TRUE</span>,</span>
<span id="cb368-5"><a href="cm1-and-cm2.html#cb368-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes =</span> cm1_selected_genes</span>
<span id="cb368-6"><a href="cm1-and-cm2.html#cb368-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/cm1_consensus_heatmap-1.png" width="672" /></p>
</div>
<div id="cm2" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> CM2</h2>
</div>
<div id="cm2-colon" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> CM2 colon</h2>
<div class="sourceCode" id="cb370"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb370-1"><a href="cm1-and-cm2.html#cb370-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running data with CopyKit for the primary colon sample</span></span>
<span id="cb370-2"><a href="cm1-and-cm2.html#cb370-2" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">runVarbin</span>(<span class="st">&quot;/mnt/lab/users/dminussi/projects/CopyKit_Manuscript_Code/datasets/CM2/colon/marked_bams&quot;</span>, <span class="at">remove_Y =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Counting reads for genome hg38 and resolution: 220kb</code></pre>
<pre><code>## 89 bam files had less than 10 mean bincounts and were removed.</code></pre>
<pre><code>## Performing GC correction.</code></pre>
<pre><code>## Smoothing bin counts.</code></pre>
<pre><code>## Running segmentation algorithm: CBS for genome hg38</code></pre>
<pre><code>## Merging levels.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb378"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb378-1"><a href="cm1-and-cm2.html#cb378-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding diploid and low-quality cells and excluding it from the copykit object</span></span>
<span id="cb378-2"><a href="cm1-and-cm2.html#cb378-2" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findOutliers</span>(cm2_tumor_primary)</span></code></pre></div>
<pre><code>## Calculating correlation matrix.</code></pre>
<pre><code>## Marked 119 cells as outliers.</code></pre>
<pre><code>## Adding information to metadata. Access with colData(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb383"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb383-1"><a href="cm1-and-cm2.html#cb383-1" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findAneuploidCells</span>(cm2_tumor_primary)</span></code></pre></div>
<pre><code>## number of iterations= 8</code></pre>
<pre><code>## Copykit detected 8 that are possibly diploid cells using a resolution of: 0.051</code></pre>
<pre><code>## Added information to colData(CopyKit).</code></pre>
<div class="sourceCode" id="cb387"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb387-1"><a href="cm1-and-cm2.html#cb387-1" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> cm2_tumor_primary[,<span class="fu">colData</span>(cm2_tumor_primary)<span class="sc">$</span>outlier <span class="sc">==</span> <span class="cn">FALSE</span>]</span>
<span id="cb387-2"><a href="cm1-and-cm2.html#cb387-2" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> cm2_tumor_primary[,<span class="fu">colData</span>(cm2_tumor_primary)<span class="sc">$</span>is_aneuploid <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb387-3"><a href="cm1-and-cm2.html#cb387-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb387-4"><a href="cm1-and-cm2.html#cb387-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding the tissue information to colData</span></span>
<span id="cb387-5"><a href="cm1-and-cm2.html#cb387-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm2_tumor_primary)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span> <span class="st">&#39;primary&#39;</span></span></code></pre></div>
</div>
<div id="cm2-liver" class="section level2" number="4.7">
<h2><span class="header-section-number">4.7</span> CM2 liver</h2>
<div class="sourceCode" id="cb388"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb388-1"><a href="cm1-and-cm2.html#cb388-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running data with CopyKit for the liver sample</span></span>
<span id="cb388-2"><a href="cm1-and-cm2.html#cb388-2" aria-hidden="true" tabindex="-1"></a>cm2_tumor_met <span class="ot">&lt;-</span></span>
<span id="cb388-3"><a href="cm1-and-cm2.html#cb388-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">runVarbin</span>(</span>
<span id="cb388-4"><a href="cm1-and-cm2.html#cb388-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;/mnt/lab/users/dminussi/projects/CopyKit_Manuscript_Code/datasets/CM2/liver/marked_bams&quot;</span>,</span>
<span id="cb388-5"><a href="cm1-and-cm2.html#cb388-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">remove_Y =</span> <span class="cn">TRUE</span></span>
<span id="cb388-6"><a href="cm1-and-cm2.html#cb388-6" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## Counting reads for genome hg38 and resolution: 220kb</code></pre>
<pre><code>## 149 bam files had less than 10 mean bincounts and were removed.</code></pre>
<pre><code>## Performing GC correction.</code></pre>
<pre><code>## Smoothing bin counts.</code></pre>
<pre><code>## Running segmentation algorithm: CBS for genome hg38</code></pre>
<pre><code>## Merging levels.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb396"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb396-1"><a href="cm1-and-cm2.html#cb396-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Finding diploid and low-quality cells and excluding it from the copykit object</span></span>
<span id="cb396-2"><a href="cm1-and-cm2.html#cb396-2" aria-hidden="true" tabindex="-1"></a>cm2_tumor_met <span class="ot">&lt;-</span> <span class="fu">findOutliers</span>(cm2_tumor_met)</span></code></pre></div>
<pre><code>## Calculating correlation matrix.</code></pre>
<pre><code>## Marked 239 cells as outliers.</code></pre>
<pre><code>## Adding information to metadata. Access with colData(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb401"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb401-1"><a href="cm1-and-cm2.html#cb401-1" aria-hidden="true" tabindex="-1"></a>cm2_tumor_met <span class="ot">&lt;-</span> <span class="fu">findAneuploidCells</span>(cm2_tumor_met)</span></code></pre></div>
<pre><code>## number of iterations= 8</code></pre>
<pre><code>## Copykit detected 6 that are possibly diploid cells using a resolution of: 0.051</code></pre>
<pre><code>## Added information to colData(CopyKit).</code></pre>
<div class="sourceCode" id="cb405"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb405-1"><a href="cm1-and-cm2.html#cb405-1" aria-hidden="true" tabindex="-1"></a>cm2_tumor_met <span class="ot">&lt;-</span></span>
<span id="cb405-2"><a href="cm1-and-cm2.html#cb405-2" aria-hidden="true" tabindex="-1"></a>  cm2_tumor_met[, <span class="fu">colData</span>(cm2_tumor_met)<span class="sc">$</span>outlier <span class="sc">==</span> <span class="cn">FALSE</span>]</span>
<span id="cb405-3"><a href="cm1-and-cm2.html#cb405-3" aria-hidden="true" tabindex="-1"></a>cm2_tumor_met <span class="ot">&lt;-</span></span>
<span id="cb405-4"><a href="cm1-and-cm2.html#cb405-4" aria-hidden="true" tabindex="-1"></a>  cm2_tumor_met[, <span class="fu">colData</span>(cm2_tumor_met)<span class="sc">$</span>is_aneuploid <span class="sc">==</span> <span class="cn">TRUE</span>]</span>
<span id="cb405-5"><a href="cm1-and-cm2.html#cb405-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb405-6"><a href="cm1-and-cm2.html#cb405-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding the tissue information to colData</span></span>
<span id="cb405-7"><a href="cm1-and-cm2.html#cb405-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm2_tumor_met)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span> <span class="st">&#39;metastasis&#39;</span></span></code></pre></div>
</div>
<div id="cm2-merging" class="section level2" number="4.8">
<h2><span class="header-section-number">4.8</span> CM2 merging</h2>
<div class="sourceCode" id="cb406"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb406-1"><a href="cm1-and-cm2.html#cb406-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merging the three copykit objects</span></span>
<span id="cb406-2"><a href="cm1-and-cm2.html#cb406-2" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> <span class="fu">cbind</span>(cm2_tumor_primary,</span>
<span id="cb406-3"><a href="cm1-and-cm2.html#cb406-3" aria-hidden="true" tabindex="-1"></a>                    cm2_tumor_met)</span>
<span id="cb406-4"><a href="cm1-and-cm2.html#cb406-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb406-5"><a href="cm1-and-cm2.html#cb406-5" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(cm2_merged)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb411"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb411-1"><a href="cm1-and-cm2.html#cb411-1" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(cm2_merged)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 38 with median jaccard similarity of: 0.922</code></pre>
<div class="sourceCode" id="cb415"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb415-1"><a href="cm1-and-cm2.html#cb415-1" aria-hidden="true" tabindex="-1"></a>cm2_merged_suggestedk <span class="ot">&lt;-</span> <span class="fu">plotSuggestedK</span>(cm2_merged)</span>
<span id="cb415-2"><a href="cm1-and-cm2.html#cb415-2" aria-hidden="true" tabindex="-1"></a>cm2_merged_suggestedk</span></code></pre></div>
<p><img src="_main_files/figure-html/cm2_merge-1.png" width="672" /></p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb416-1"><a href="cm1-and-cm2.html#cb416-1" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(cm2_merged)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 38</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 70 outliers cells (group &#39;c0&#39;)</code></pre>
<pre><code>## Found 11 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb422"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb422-1"><a href="cm1-and-cm2.html#cb422-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HDBSCAN is an outlier aware clustering algorithm</span></span>
<span id="cb422-2"><a href="cm1-and-cm2.html#cb422-2" aria-hidden="true" tabindex="-1"></a><span class="co"># in this analysis all cells marked as outliers (c0) from hdbscan are excluded.</span></span>
<span id="cb422-3"><a href="cm1-and-cm2.html#cb422-3" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> cm2_merged[, <span class="fu">colData</span>(cm2_merged)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span>
<span id="cb422-4"><a href="cm1-and-cm2.html#cb422-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-5"><a href="cm1-and-cm2.html#cb422-5" aria-hidden="true" tabindex="-1"></a>cm2_merged_tp_umap_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(cm2_merged, <span class="at">label =</span> <span class="st">&#39;timepoint&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: timepoint.</code></pre>
<div class="sourceCode" id="cb425"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb425-1"><a href="cm1-and-cm2.html#cb425-1" aria-hidden="true" tabindex="-1"></a>cm2_merged_tp_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/cm2_merge-2.png" width="672" /></p>
<div class="sourceCode" id="cb426"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb426-1"><a href="cm1-and-cm2.html#cb426-1" aria-hidden="true" tabindex="-1"></a>cm2_merged_umap_p <span class="ot">&lt;-</span> <span class="fu">plotUmap</span>(cm2_merged, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: subclones.</code></pre>
<div class="sourceCode" id="cb429"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb429-1"><a href="cm1-and-cm2.html#cb429-1" aria-hidden="true" tabindex="-1"></a>cm2_merged_umap_p</span></code></pre></div>
<p><img src="_main_files/figure-html/cm2_merge-3.png" width="672" /></p>
<div class="sourceCode" id="cb430"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb430-1"><a href="cm1-and-cm2.html#cb430-1" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(cm2_merged)</span>
<span id="cb430-2"><a href="cm1-and-cm2.html#cb430-2" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> <span class="fu">runConsensusPhylo</span>(cm2_merged)</span>
<span id="cb430-3"><a href="cm1-and-cm2.html#cb430-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb430-4"><a href="cm1-and-cm2.html#cb430-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm2_merged)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span></span>
<span id="cb430-5"><a href="cm1-and-cm2.html#cb430-5" aria-hidden="true" tabindex="-1"></a>  forcats<span class="sc">::</span><span class="fu">fct_relevel</span>(<span class="fu">colData</span>(cm2_merged)<span class="sc">$</span>timepoint,</span>
<span id="cb430-6"><a href="cm1-and-cm2.html#cb430-6" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">c</span>(<span class="st">&quot;primary&quot;</span>, <span class="st">&quot;metastasis&quot;</span>))</span>
<span id="cb430-7"><a href="cm1-and-cm2.html#cb430-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb430-8"><a href="cm1-and-cm2.html#cb430-8" aria-hidden="true" tabindex="-1"></a>cm2_selected_genes <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb430-9"><a href="cm1-and-cm2.html#cb430-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;IGF1R&quot;</span>,</span>
<span id="cb430-10"><a href="cm1-and-cm2.html#cb430-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;SMAD6&quot;</span>,</span>
<span id="cb430-11"><a href="cm1-and-cm2.html#cb430-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;FGFR2&quot;</span>,</span>
<span id="cb430-12"><a href="cm1-and-cm2.html#cb430-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;STAT3&quot;</span>,</span>
<span id="cb430-13"><a href="cm1-and-cm2.html#cb430-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;SOX9&quot;</span>,</span>
<span id="cb430-14"><a href="cm1-and-cm2.html#cb430-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;KLF5&quot;</span>,</span>
<span id="cb430-15"><a href="cm1-and-cm2.html#cb430-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;FGF9&quot;</span>,</span>
<span id="cb430-16"><a href="cm1-and-cm2.html#cb430-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;MSH2&quot;</span>,</span>
<span id="cb430-17"><a href="cm1-and-cm2.html#cb430-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;APC&quot;</span>,</span>
<span id="cb430-18"><a href="cm1-and-cm2.html#cb430-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb430-19"><a href="cm1-and-cm2.html#cb430-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;TNFRSF6B&quot;</span>,</span>
<span id="cb430-20"><a href="cm1-and-cm2.html#cb430-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;CHEK2&quot;</span>,</span>
<span id="cb430-21"><a href="cm1-and-cm2.html#cb430-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;MAP3K8&quot;</span></span>
<span id="cb430-22"><a href="cm1-and-cm2.html#cb430-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb430-23"><a href="cm1-and-cm2.html#cb430-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb430-24"><a href="cm1-and-cm2.html#cb430-24" aria-hidden="true" tabindex="-1"></a>cm2_merged_selected_hvg_gc <span class="ot">&lt;-</span></span>
<span id="cb430-25"><a href="cm1-and-cm2.html#cb430-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotGeneCopy</span>(cm2_merged,</span>
<span id="cb430-26"><a href="cm1-and-cm2.html#cb430-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">genes =</span> cm2_selected_genes,</span>
<span id="cb430-27"><a href="cm1-and-cm2.html#cb430-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">label =</span> <span class="st">&#39;timepoint&#39;</span>,</span>
<span id="cb430-28"><a href="cm1-and-cm2.html#cb430-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">dodge.width =</span> .<span class="dv">8</span>) <span class="sc">+</span> <span class="fu">scale_fill_hue</span>(<span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Coloring by: timepoint</code></pre>
<pre><code>## Warning in find_scaffold_genes(scCNA, genes = genes): Genes: TNFRSF6B are in excluded regions of the Varbin
## pipeline and can&#39;t be plotted.</code></pre>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which will replace the existing
## scale.</code></pre>
<div class="sourceCode" id="cb434"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb434-1"><a href="cm1-and-cm2.html#cb434-1" aria-hidden="true" tabindex="-1"></a>cm2_merged_selected_hvg_gc</span></code></pre></div>
<p><img src="_main_files/figure-html/cm2_merge-4.png" width="672" /></p>
<p>To root the tree, we will use an inferred Most Recent Common Ancestral from the primary tumor and provide that as an argument to the runConsensusPhylo function.
This consensus tree will be used by plotHeatmap to order the subclones on the plot</p>
<div class="sourceCode" id="cb435"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb435-1"><a href="cm1-and-cm2.html#cb435-1" aria-hidden="true" tabindex="-1"></a><span class="co"># clustering and inferring the MRCA from the clusters</span></span>
<span id="cb435-2"><a href="cm1-and-cm2.html#cb435-2" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(cm2_tumor_primary)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb440"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb440-1"><a href="cm1-and-cm2.html#cb440-1" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(cm2_tumor_primary)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 28 with median jaccard similarity of: 0.944</code></pre>
<div class="sourceCode" id="cb444"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb444-1"><a href="cm1-and-cm2.html#cb444-1" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(cm2_tumor_primary)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 28</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 17 outliers cells (group &#39;c0&#39;)</code></pre>
<pre><code>## Found 8 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<div class="sourceCode" id="cb450"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb450-1"><a href="cm1-and-cm2.html#cb450-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HDBSCAN is an outlier aware clustering algorithm</span></span>
<span id="cb450-2"><a href="cm1-and-cm2.html#cb450-2" aria-hidden="true" tabindex="-1"></a><span class="co"># in this analysis all cells marked as outliers (c0) from hdbscan are excluded.</span></span>
<span id="cb450-3"><a href="cm1-and-cm2.html#cb450-3" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span></span>
<span id="cb450-4"><a href="cm1-and-cm2.html#cb450-4" aria-hidden="true" tabindex="-1"></a>  cm2_tumor_primary[, <span class="fu">colData</span>(cm2_tumor_primary)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span>
<span id="cb450-5"><a href="cm1-and-cm2.html#cb450-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb450-6"><a href="cm1-and-cm2.html#cb450-6" aria-hidden="true" tabindex="-1"></a><span class="co"># calculating the consensus of the Merged dataset and using the inferred</span></span>
<span id="cb450-7"><a href="cm1-and-cm2.html#cb450-7" aria-hidden="true" tabindex="-1"></a><span class="co"># primary MRCA as the root of the tree</span></span>
<span id="cb450-8"><a href="cm1-and-cm2.html#cb450-8" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(cm2_tumor_primary)</span>
<span id="cb450-9"><a href="cm1-and-cm2.html#cb450-9" aria-hidden="true" tabindex="-1"></a>cm2_tumor_primary <span class="ot">&lt;-</span> <span class="fu">inferMrca</span>(cm2_tumor_primary, <span class="at">value =</span> <span class="fl">0.8</span>)</span>
<span id="cb450-10"><a href="cm1-and-cm2.html#cb450-10" aria-hidden="true" tabindex="-1"></a>cm2_merged <span class="ot">&lt;-</span> <span class="fu">runConsensusPhylo</span>(</span>
<span id="cb450-11"><a href="cm1-and-cm2.html#cb450-11" aria-hidden="true" tabindex="-1"></a>  cm2_merged,</span>
<span id="cb450-12"><a href="cm1-and-cm2.html#cb450-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">root =</span> <span class="st">&#39;user&#39;</span>,</span>
<span id="cb450-13"><a href="cm1-and-cm2.html#cb450-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">root_user =</span> <span class="fu">metadata</span>(cm2_tumor_primary)<span class="sc">$</span>inferred_mrca</span>
<span id="cb450-14"><a href="cm1-and-cm2.html#cb450-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb450-15"><a href="cm1-and-cm2.html#cb450-15" aria-hidden="true" tabindex="-1"></a><span class="fu">consensusPhylo</span>(cm2_merged) <span class="ot">&lt;-</span> phytools<span class="sc">::</span><span class="fu">rotateNodes</span>(<span class="fu">consensusPhylo</span>(cm2_merged), <span class="fu">c</span>(<span class="dv">14</span>))</span>
<span id="cb450-16"><a href="cm1-and-cm2.html#cb450-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb450-17"><a href="cm1-and-cm2.html#cb450-17" aria-hidden="true" tabindex="-1"></a><span class="co"># relevel factors to plot in the desired order</span></span>
<span id="cb450-18"><a href="cm1-and-cm2.html#cb450-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(cm2_merged)<span class="sc">$</span>timepoint <span class="ot">&lt;-</span></span>
<span id="cb450-19"><a href="cm1-and-cm2.html#cb450-19" aria-hidden="true" tabindex="-1"></a>  forcats<span class="sc">::</span><span class="fu">fct_relevel</span>(<span class="fu">colData</span>(cm2_merged)<span class="sc">$</span>timepoint,</span>
<span id="cb450-20"><a href="cm1-and-cm2.html#cb450-20" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">c</span>(<span class="st">&quot;metastasis&quot;</span>, <span class="st">&quot;primary&quot;</span>))</span>
<span id="cb450-21"><a href="cm1-and-cm2.html#cb450-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb450-22"><a href="cm1-and-cm2.html#cb450-22" aria-hidden="true" tabindex="-1"></a><span class="co"># plotting the phylogeny with subclones labels and pie charts</span></span>
<span id="cb450-23"><a href="cm1-and-cm2.html#cb450-23" aria-hidden="true" tabindex="-1"></a><span class="co">#indicating the frequency of each timepoint</span></span>
<span id="cb450-24"><a href="cm1-and-cm2.html#cb450-24" aria-hidden="true" tabindex="-1"></a>cm2_merged_consensus_phylo <span class="ot">&lt;-</span></span>
<span id="cb450-25"><a href="cm1-and-cm2.html#cb450-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotPhylo</span>(cm2_merged,</span>
<span id="cb450-26"><a href="cm1-and-cm2.html#cb450-26" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>,</span>
<span id="cb450-27"><a href="cm1-and-cm2.html#cb450-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">consensus =</span> <span class="cn">TRUE</span>,</span>
<span id="cb450-28"><a href="cm1-and-cm2.html#cb450-28" aria-hidden="true" tabindex="-1"></a>            <span class="at">group =</span> <span class="st">&#39;timepoint&#39;</span>)</span>
<span id="cb450-29"><a href="cm1-and-cm2.html#cb450-29" aria-hidden="true" tabindex="-1"></a>cm2_merged_consensus_phylo</span></code></pre></div>
<p><img src="_main_files/figure-html/cm2_tree-1.png" width="672" /></p>
<div class="sourceCode" id="cb451"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb451-1"><a href="cm1-and-cm2.html#cb451-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(cm2_merged, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>, <span class="st">&#39;timepoint&#39;</span>))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/cm2_heatmap-1.png" width="768" /></p>
<div class="sourceCode" id="cb453"><pre class="sourceCode r bg-success"><code class="sourceCode r"><span id="cb453-1"><a href="cm1-and-cm2.html#cb453-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb453-2"><a href="cm1-and-cm2.html#cb453-2" aria-hidden="true" tabindex="-1"></a>  cm2_merged,</span>
<span id="cb453-3"><a href="cm1-and-cm2.html#cb453-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&#39;subclones&#39;</span>),</span>
<span id="cb453-4"><a href="cm1-and-cm2.html#cb453-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">consensus =</span> <span class="cn">TRUE</span>,</span>
<span id="cb453-5"><a href="cm1-and-cm2.html#cb453-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes =</span> cm2_selected_genes</span>
<span id="cb453-6"><a href="cm1-and-cm2.html#cb453-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Warning in find_scaffold_genes(scCNA, genes): Genes: TNFRSF6B are in excluded regions of the Varbin pipeline
## and can&#39;t be plotted.</code></pre>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="_main_files/figure-html/cm2_consensus_heatmap-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bm1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/navinlabcode/copykit/blob/master/04-CM1_CM2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
